--- Makefile.original	2020-10-04 23:47:50.000000000 +0000
+++ Makefile	2020-10-05 00:36:38.406462000 +0000
@@ -1,6 +1,3 @@
-# ordena
-#-----------------------------
-ORDENA := $(shell hostname)
 
 
 
@@ -19,75 +16,15 @@
 
 # main path
 #-----------------------------
-#ifeq ($(ORDENA),pf-070210dl)
-#  MAIN_PATH=/scr2_pf-070210dl/cabr_ju/ross/blind/asalto26.5
-ifeq ($(ORDENA),pf-070210dl.intra.dlr.de)
-  MAIN_PATH=/scr2/cabr_ju/ross/blind/asalto26.5
-else
-ifeq ($(ORDENA),OTS44)
-  MAIN_PATH=/data/COROT/qmon/asalto26.5
-else
-ifeq ($(ORDENA),canijo)
-  MAIN_PATH=/datos/blind/asalto26.5
-else
-ifeq ($(ORDENA),stahl)
-  MAIN_PATH=/home/juano/blind/asalto26.5
-else
-ifeq ($(ORDENA),warg01)
-  MAIN_PATH=/home/warg01/cabrera/blind/asalto26.5
-else
-ifeq ($(ORDENA),warg02)
-  MAIN_PATH=/home/warg01/cabrera/blind/asalto26.5
-else
-  $(error error, computer $(ORDENA) not recognized)
-endif
-endif
-endif
-endif
-endif
-endif
-
+MAIN_PATH=.
 
 
 # cfitsio
 #-----------------------------
-ifeq ($(ORDENA),pf-070210dl.intra.dlr.de)
-  CFITSIO_PATH = /usr
-  CFITSIO_LIB_PATH = $(CFITSIO_PATH)/lib
-  CFITSIO_INC_PATH = $(CFITSIO_PATH)/include/cfitsio
-else
-ifeq ($(ORDENA),OTS44)
-  CFITSIO_PATH = /export/opt/cfitsio
-  CFITSIO_LIB_PATH = $(CFITSIO_PATH)/lib
-  CFITSIO_INC_PATH = $(CFITSIO_PATH)/include
-else
-ifeq ($(ORDENA),canijo)
-  CFITSIO_PATH = /usr
-  CFITSIO_LIB_PATH = $(CFITSIO_PATH)/lib/x86_64-linux-gnu
-  CFITSIO_INC_PATH = $(CFITSIO_PATH)/include
-else
-ifeq ($(ORDENA),stahl)
-  CFITSIO_PATH = /usr
-  CFITSIO_LIB_PATH = $(CFITSIO_PATH)/lib/x86_64-linux-gnu
-  CFITSIO_INC_PATH = $(CFITSIO_PATH)/include
-else
-ifeq ($(ORDENA),warg01)
-  CFITSIO_PATH = /usr
-  CFITSIO_LIB_PATH = $(CFITSIO_PATH)/lib/x86_64-linux-gnu
-  CFITSIO_INC_PATH = $(CFITSIO_PATH)/include
-else
-ifeq ($(ORDENA),warg02)
-  CFITSIO_PATH = /usr
-  CFITSIO_LIB_PATH = $(CFITSIO_PATH)/lib/x86_64-linux-gnu
-  CFITSIO_INC_PATH = $(CFITSIO_PATH)/include
-else
-  $(error error, computer $(ORDENA) not recognized)
-endif
-endif
-endif
-endif
-endif
-endif
+CFITSIO_PATH = .
+CFITSIO_LIB_PATH = .
+CFITSIO_INC_PATH = .
+
 #--
 CFITSIO_INC = -I $(CFITSIO_INC_PATH)
 CFITSIO_LNK = -L$(CFITSIO_LIB_PATH) -lcfitsio -lm
@@ -96,43 +33,10 @@
 
 # GSL
 #-----------------------------
-ifeq ($(ORDENA),pf-070210dl.intra.dlr.de)
-  GSL_PATH=/usr
-  GSL_LIB_PATH=$(GSL_PATH)/lib64
-  GSL_INC_PATH=$(GSL_PATH)/include/gsl
-else
-ifeq ($(ORDENA),OTS44)
-  GSL_PATH=/opt
-  GSL_LIB_PATH=$(GSL_PATH)/lib
-  GSL_INC_PATH=$(GSL_PATH)/include/gsl
-else
-ifeq ($(ORDENA),canijo)
-  GSL_PATH=/usr
-  GSL_LIB_PATH=$(GSL_PATH)/lib
-  GSL_INC_PATH=$(GSL_PATH)/include/gsl
-else
-ifeq ($(ORDENA),stahl)
-  GSL_PATH=/usr
-  GSL_LIB_PATH=$(GSL_PATH)/lib
-  GSL_INC_PATH=$(GSL_PATH)/include/gsl
-else
-ifeq ($(ORDENA),warg01)
-  GSL_PATH=/usr
-  GSL_LIB_PATH=$(GSL_PATH)/lib
-  GSL_INC_PATH=$(GSL_PATH)/include/gsl
-else
-ifeq ($(ORDENA),warg02)
-  GSL_PATH=/usr
-  GSL_LIB_PATH=$(GSL_PATH)/lib
-  GSL_INC_PATH=$(GSL_PATH)/include/gsl
-else
-  $(error error, computer $(ORDENA) not recognized)
-endif
-endif
-endif
-endif
-endif
-endif
+GSL_PATH=.
+GSL_LIB_PATH=.
+GSL_INC_PATH=.
+
 #--
 GSL_INC = -I $(GSL_INC_PATH)
 GSL_LNK = -L$(GSL_LIB_PATH) -lgsl -lgslcblas -lm
@@ -141,48 +45,15 @@
 
 # libjuan
 #-----------------------------
-ifeq ($(ORDENA),pf-070210dl.intra.dlr.de)
-  JUAN_PATH=/scr2/cabr_ju/ross/libC
-  JUAN_LIB_PATH=$(JUAN_PATH)/lib/$(ORDENA)
-  JUAN_INC_PATH=$(JUAN_PATH)
-else
-ifeq ($(ORDENA),OTS44)
-  JUAN_PATH=/data/COROT/qmon/libC
-  JUAN_LIB_PATH=$(JUAN_PATH)/lib/$(ORDENA)
-  JUAN_INC_PATH=$(JUAN_PATH)
-else
-ifeq ($(ORDENA),canijo)
-  JUAN_PATH=/datos/blind/libC
-  JUAN_LIB_PATH=$(JUAN_PATH)/lib/$(ORDENA)
-  JUAN_INC_PATH=$(JUAN_PATH)
-else
-ifeq ($(ORDENA),stahl)
-  JUAN_PATH=/home/juano/blind/libC
-  JUAN_LIB_PATH=$(JUAN_PATH)/lib/$(ORDENA)
-  JUAN_INC_PATH=$(JUAN_PATH)
-else
-ifeq ($(ORDENA),warg01)
-  JUAN_PATH=/home/warg01/cabrera/libC
-  JUAN_LIB_PATH=$(JUAN_PATH)/lib/$(ORDENA)
-  JUAN_INC_PATH=$(JUAN_PATH)
-else
-ifeq ($(ORDENA),warg02)
-  JUAN_PATH=/home/warg01/cabrera/libC
-  JUAN_LIB_PATH=$(JUAN_PATH)/lib/$(ORDENA)
-  JUAN_INC_PATH=$(JUAN_PATH)
-else
-  $(error error, computer $(ORDENA) not recognized)
-endif
-endif
-endif
-endif
-endif
-endif
+JUAN_PATH=../juan
+JUAN_LIB_PATH=$(JUAN_PATH)/lib
+JUAN_INC_PATH=$(JUAN_PATH)
+
 #--
 JUAN_CLAU = juan
 JUAN_HDR = $(JUAN_PATH)/$(JUAN_CLAU).h
 JUAN_SRC = $(JUAN_PATH)/$(JUAN_CLAU).c
-JUAN_OBJ = $(JUAN_PATH)/obj/$(ORDENA)/$(JUAN_CLAU).o
+JUAN_OBJ = $(JUAN_PATH)/obj/$(JUAN_CLAU).o
 JUAN_LIB = $(JUAN_LIB_PATH)/lib$(JUAN_CLAU).a
 JUAN_SLS = $(JUAN_SRC) $(JUAN_HDR)
 JUAN_OLS = $(JUAN_OBJ)
@@ -194,49 +65,17 @@
 
 # libjuan_fits
 #-----------------------------
-ifeq ($(ORDENA),pf-070210dl.intra.dlr.de)
-  JUANFITS_PATH=/scr2/cabr_ju/ross/libC
-  JUANFITS_LIB_PATH=$(JUANFITS_PATH)/lib/$(ORDENA)
-  JUANFITS_INC_PATH=$(JUANFITS_PATH)
-else
-ifeq ($(ORDENA),OTS44)
-  JUANFITS_PATH=/data/COROT/qmon/libC
-  JUANFITS_LIB_PATH=$(JUANFITS_PATH)/lib/$(ORDENA)
-  JUANFITS_INC_PATH=$(JUANFITS_PATH)
-else
-ifeq ($(ORDENA),canijo)
-  JUANFITS_PATH=/datos/blind/libC
-  JUANFITS_LIB_PATH=$(JUANFITS_PATH)/lib/$(ORDENA)
-  JUANFITS_INC_PATH=$(JUANFITS_PATH)
-else
-ifeq ($(ORDENA),stahl)
-  JUANFITS_PATH=/home/juano/blind/libC
-  JUANFITS_LIB_PATH=$(JUANFITS_PATH)/lib/$(ORDENA)
-  JUANFITS_INC_PATH=$(JUANFITS_PATH)
-else
-ifeq ($(ORDENA),warg01)
-  JUANFITS_PATH=/home/warg01/cabrera/libC
-  JUANFITS_LIB_PATH=$(JUANFITS_PATH)/lib/$(ORDENA)
-  JUANFITS_INC_PATH=$(JUANFITS_PATH)
-else
-ifeq ($(ORDENA),warg02)
-  JUANFITS_PATH=/home/warg01/cabrera/libC
-  JUANFITS_LIB_PATH=$(JUANFITS_PATH)/lib/$(ORDENA)
-  JUANFITS_INC_PATH=$(JUANFITS_PATH)
-else
-  $(error error, computer $(ORDENA) not recognized)
-endif
-endif
-endif
-endif
-endif
-endif
+JUAN_PATH=../juan
+JUAN_LIB_PATH=$(JUAN_PATH)/lib
+JUAN_INC_PATH=$(JUAN_PATH)
+
 #--
 JUANFITS_CLAU = juan_fits
+JUANFITS_PATH = ../juan
 JUANFITS_HDR = $(JUANFITS_PATH)/$(JUANFITS_CLAU).h
 JUANFITS_SRC = $(JUANFITS_PATH)/$(JUANFITS_CLAU).c
-JUANFITS_OBJ = $(JUANFITS_PATH)/obj/$(ORDENA)/$(JUANFITS_CLAU).o
-JUANFITS_LIB = $(JUANFITS_LIB_PATH)/lib$(JUANFITS_CLAU).a
+JUANFITS_OBJ = $(JUANFITS_PATH)/obj/$(JUANFITS_CLAU).o
+JUANFITS_LIB = $(JUANFITS_PATH)/lib$(JUANFITS_CLAU).a
 JUANFITS_SLS = $(JUANFITS_SRC) $(JUANFITS_HDR)
 JUANFITS_OLS = $(JUANFITS_OBJ)
 JUANFITS_LLS = $(JUANFITS_LIB)
@@ -251,13 +90,13 @@
 BT2_PATH = $(MAIN_PATH)
 BT2_HDR = $(BT2_PATH)/src/$(BT2_CLAU).h
 BT2_SRC = $(BT2_PATH)/src/$(BT2_CLAU).c
-BT2_OBJ = $(BT2_PATH)/obj/$(ORDENA)/$(BT2_CLAU).o
-BT2_LIB = $(BT2_PATH)/lib/$(ORDENA)/lib$(BT2_CLAU).a
+BT2_OBJ = $(BT2_PATH)/obj/$(BT2_CLAU).o
+BT2_LIB = $(BT2_PATH)/lib/lib$(BT2_CLAU).a
 BT2_SLS = $(BT2_SRC) $(BT2_HDR) $(JUAN_HDR) $(JUANFITS_HDR)
 BT2_OLS = $(BT2_OBJ) 
 BT2_LLS = $(BT2_LIB) $(JUAN_LIB) $(JUANFITS_LIB)
 BT2_INC = -I$(BT2_PATH)/src $(JUAN_INC) $(JUANFITS_INC) 
-BT2_LNK = -L$(BT2_PATH)/lib/$(ORDENA) -l$(BT2_CLAU) $(JUAN_LNK) $(JUANFITS_LNK) 
+BT2_LNK = -L$(BT2_PATH)/lib -l$(BT2_CLAU) $(JUAN_LNK) $(JUANFITS_LNK) 
 
 
 
@@ -267,13 +106,13 @@
 BT2CLNSET_PATH = $(MAIN_PATH)
 BT2CLNSET_HDR = $(BT2CLNSET_PATH)/src/$(BT2CLNSET_CLAU).h
 BT2CLNSET_SRC = $(BT2CLNSET_PATH)/src/$(BT2CLNSET_CLAU).c
-BT2CLNSET_OBJ = $(BT2CLNSET_PATH)/obj/$(ORDENA)/$(BT2CLNSET_CLAU).o
-BT2CLNSET_LIB = $(BT2CLNSET_PATH)/lib/$(ORDENA)/lib$(BT2CLNSET_CLAU).a
+BT2CLNSET_OBJ = $(BT2CLNSET_PATH)/obj/$(BT2CLNSET_CLAU).o
+BT2CLNSET_LIB = $(BT2CLNSET_PATH)/lib/lib$(BT2CLNSET_CLAU).a
 BT2CLNSET_SLS = $(BT2CLNSET_SRC) $(BT2CLNSET_HDR) $(JUAN_HDR) $(JUANFITS_HDR)
 BT2CLNSET_OLS = $(BT2CLNSET_OBJ) 
 BT2CLNSET_LLS = $(BT2CLNSET_LIB) $(JUAN_LIB) $(JUANFITS_LIB)
 BT2CLNSET_INC = -I$(BT2CLNSET_PATH)/src $(JUAN_INC) $(JUANFITS_INC) 
-BT2CLNSET_LNK = -L$(BT2CLNSET_PATH)/lib/$(ORDENA) -l$(BT2CLNSET_CLAU) $(JUAN_LNK) $(JUANFITS_LNK) 
+BT2CLNSET_LNK = -L$(BT2CLNSET_PATH)/lib -l$(BT2CLNSET_CLAU) $(JUAN_LNK) $(JUANFITS_LNK) 
 
 
 
@@ -283,13 +122,13 @@
 BIN_PATH = $(MAIN_PATH)
 BIN_HDR = $(BIN_PATH)/src/$(BIN_CLAU).h
 BIN_SRC = $(BIN_PATH)/src/$(BIN_CLAU).c
-BIN_OBJ = $(BIN_PATH)/obj/$(ORDENA)/$(BIN_CLAU).o
-BIN_LIB = $(BIN_PATH)/lib/$(ORDENA)/lib$(BIN_CLAU).a
+BIN_OBJ = $(BIN_PATH)/obj/$(BIN_CLAU).o
+BIN_LIB = $(BIN_PATH)/lib/lib$(BIN_CLAU).a
 BIN_SLS = $(BIN_SRC) $(BIN_HDR) $(BT2_HDR)
 BIN_OLS = $(BIN_OBJ) 
 BIN_LLS = $(BIN_LIB) $(BT2_LLS)
 BIN_INC = -I$(BIN_PATH)/src $(BT2_INC)
-BIN_LNK = -L$(BIN_PATH)/lib/$(ORDENA) -l$(BIN_CLAU) $(BT2_LNK)
+BIN_LNK = -L$(BIN_PATH)/lib -l$(BIN_CLAU) $(BT2_LNK)
 
 
 
@@ -299,13 +138,13 @@
 MPL_PATH = $(MAIN_PATH)
 MPL_HDR = $(MPL_PATH)/src/$(MPL_CLAU).h
 MPL_SRC = $(MPL_PATH)/src/$(MPL_CLAU).c
-MPL_OBJ = $(MPL_PATH)/obj/$(ORDENA)/$(MPL_CLAU).o
-MPL_LIB = $(MPL_PATH)/lib/$(ORDENA)/lib$(MPL_CLAU).a
+MPL_OBJ = $(MPL_PATH)/obj/$(MPL_CLAU).o
+MPL_LIB = $(MPL_PATH)/lib/lib$(MPL_CLAU).a
 MPL_SLS = $(MPL_SRC) $(MPL_HDR) $(BIN_HDR)
 MPL_OLS = $(MPL_OBJ) 
 MPL_LLS = $(MPL_LIB) $(BIN_LLS)
 MPL_INC = -I$(MPL_PATH)/src $(BIN_INC)
-MPL_LNK = -L$(MPL_PATH)/lib/$(ORDENA) -l$(MPL_CLAU) $(BIN_LNK)
+MPL_LNK = -L$(MPL_PATH)/lib -l$(MPL_CLAU) $(BIN_LNK)
 
 
 
@@ -315,8 +154,8 @@
 TMG_PATH = $(MAIN_PATH)
 TMG_HDR = $(TMG_PATH)/src/$(TMG_CLAU).h
 TMG_SRC = $(TMG_PATH)/src/$(TMG_CLAU).c
-TMG_OBJ = $(TMG_PATH)/obj/$(ORDENA)/$(TMG_CLAU).o
-TMG_LIB = $(TMG_PATH)/lib/$(ORDENA)/lib$(TMG_CLAU).a
+TMG_OBJ = $(TMG_PATH)/obj/$(TMG_CLAU).o
+TMG_LIB = $(TMG_PATH)/lib/lib$(TMG_CLAU).a
 TMG_SLS = $(TMP_SRC) $(TMG_HDR) $(MPL_HDR) $(BIN_HDR)
 TMG_OLS = $(TMG_OBJ) 
 TMG_LLS = $(TMG_LIB) $(MPL_LLS)
@@ -330,7 +169,6 @@
 all : onyva onyva_imagette onyva_kepler onyva_kepler_eduardo onyva_k2vanderburg onyva_everest onyva_k2sc onyva_polar onyva_tess onyva_cheops onyva_tasoc onyva_clnset tsevi AN2_onyva winid header lector_ia binarias binarias_fake timing timing_ajtlin timing_irwin AN2_timing fitmodpl tendencia fold traduce rednoise readclnset hsa
 
 test :
-	@echo $(ORDENA)
 	@echo $(JUAN_PATH)
 	@echo $(JUANFITS_PATH)
 	@echo $(GSL_PATH)
@@ -340,13 +178,15 @@
          onyva onyva_imagette onyva_kepler onyva_kepler_eduardo onyva_k2vanderburg onyva_everest onyva_k2sc onyva_polar onyva_tess onyva_cheops onyva_tasoc onyva_clnset tsevi AN2_onyva winid header lector_ia binarias binarias_fake timing timing_ajtlin timing_irwin AN2_timing fitmodpl datejd tendencia fold traduce horriblesospecha rednoise wdscdb readclnset hsa
 
 
+make_directories:
+	mkdir -p ../juan/obj ../juan/lib ../asalto26.5/obj ../asalto26.5/bin ../asalto26.5/lib ../asalto27/obj ../asalto27/bin ../asalto27/lib
 
 # onyva
 #-----------------------------
 ONYVA_CLAU = onyva
 ONYVA_SRC = $(MAIN_PATH)/src/$(ONYVA_CLAU).c
-ONYVA_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(ONYVA_CLAU).o
-ONYVA_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(ONYVA_CLAU).exe
+ONYVA_OBJ = $(MAIN_PATH)/obj/$(ONYVA_CLAU).o $(JUANFITS_OBJ)
+ONYVA_EXE = $(MAIN_PATH)/bin/$(ONYVA_CLAU).exe
 ONYVA_SLS = $(ONYVA_SRC) $(BT2_HDR)
 ONYVA_OLS = $(ONYVA_OBJ) 
 ONYVA_LLS = $(BT2_LLS)
@@ -356,12 +196,12 @@
 $(ONYVA_CLAU) : $(ONYVA_EXE)
 	@echo " clau: $(ONYVA_CLAU)"
 
-$(ONYVA_EXE) : $(ONYVA_OLS) $(ONYVA_LLS)
+$(ONYVA_EXE) : $(ONYVA_OLS) $(ONYVA_LLS) make_directories
 	@echo "****** linking $(ONYVA_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) $(ONYVA_OLS) -o $@ $(ONYVA_LNK) 
 	@echo "## $(ONYVA_CLAU) linked successfully"
 
-$(ONYVA_OBJ) : $(ONYVA_SLS)
+$(ONYVA_OBJ) : $(ONYVA_SLS) make_directories
 	@echo "****** compiling $(ONYVA_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(ONYVA_SRC) $(ONYVA_INC) -o $@ 
 	@echo "## $(ONYVA_CLAU) compiled successfully"
@@ -372,8 +212,8 @@
 #-----------------------------
 ONYVAIMAG_CLAU = onyva_imagette
 ONYVAIMAG_SRC = $(MAIN_PATH)/src/$(ONYVAIMAG_CLAU).c
-ONYVAIMAG_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(ONYVAIMAG_CLAU).o
-ONYVAIMAG_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(ONYVAIMAG_CLAU).exe
+ONYVAIMAG_OBJ = $(MAIN_PATH)/obj/$(ONYVAIMAG_CLAU).o $(JUANFITS_OBJ)
+ONYVAIMAG_EXE = $(MAIN_PATH)/bin/$(ONYVAIMAG_CLAU).exe
 ONYVAIMAG_SLS = $(ONYVAIMAG_SRC) $(ONYVAIMAG_HDR) $(BT2_HDR)
 ONYVAIMAG_OLS = $(ONYVAIMAG_OBJ)
 ONYVAIMAG_LLS = $(BT2_LLS)
@@ -384,12 +224,12 @@
 $(ONYVAIMAG_CLAU) : $(ONYVAIMAG_EXE)
 	@echo " clau: $(ONYVAIMAG_CLAU)"
 
-$(ONYVAIMAG_EXE) : $(ONYVAIMAG_OLS) $(ONYVAIMAG_LLS)
+$(ONYVAIMAG_EXE) : $(ONYVAIMAG_OLS) $(ONYVAIMAG_LLS) make_directories
 	@echo "****** linking $(ONYVAIMAG_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) $(ONYVAIMAG_OLS) -o $@ $(BT2_LNK) 
 	@echo "## $(ONYVAIMAG_CLAU) linked successfully"
 
-$(ONYVAIMAG_OBJ) : $(ONYVAIMAG_SLS)
+$(ONYVAIMAG_OBJ) : $(ONYVAIMAG_SLS) make_directories
 	@echo "****** compiling $(ONYVAIMAG_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(ONYVAIMAG_SRC) $(ONYVAIMAG_INC) -o $@  
 	@echo "## $(ONYVAIMAG_CLAU) compiled successfully"
@@ -399,8 +239,8 @@
 #-----------------------------
 ONYVAIMAGC9B_CLAU = onyva_imagette_c9b
 ONYVAIMAGC9B_SRC = $(MAIN_PATH)/src/$(ONYVAIMAGC9B_CLAU).c
-ONYVAIMAGC9B_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(ONYVAIMAGC9B_CLAU).o
-ONYVAIMAGC9B_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(ONYVAIMAGC9B_CLAU).exe
+ONYVAIMAGC9B_OBJ = $(MAIN_PATH)/obj/$(ONYVAIMAGC9B_CLAU).o
+ONYVAIMAGC9B_EXE = $(MAIN_PATH)/bin/$(ONYVAIMAGC9B_CLAU).exe
 ONYVAIMAGC9B_SLS = $(ONYVAIMAGC9B_SRC) $(ONYVAIMAGC9B_HDR) $(BT2_HDR)
 ONYVAIMAGC9B_OLS = $(ONYVAIMAGC9B_OBJ)
 ONYVAIMAGC9B_LLS = $(BT2_LLS)
@@ -409,12 +249,12 @@
 $(ONYVAIMAGC9B_CLAU) : $(ONYVAIMAGC9B_EXE)
 	@echo " clau: $(ONYVAIMAGC9B_CLAU)"
 
-$(ONYVAIMAGC9B_EXE) : $(ONYVAIMAGC9B_OLS) $(ONYVAIMAGC09B_LLS)
+$(ONYVAIMAGC9B_EXE) : $(ONYVAIMAGC9B_OLS) $(ONYVAIMAGC09B_LLS) make_directories
 	@echo "****** linking $(ONYVAIMAGC9B_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) $(ONYVAIMAGC9B_OLS) -o $@ $(ONYVAIMAGC9B_LNK) 
 	@echo "## $(ONYVAIMAGC9B_CLAU) linked successfully"
 
-$(ONYVAIMAGC9B_OBJ) : $(ONYVAIMAGC9B_SLS)
+$(ONYVAIMAGC9B_OBJ) : $(ONYVAIMAGC9B_SLS) make_directories
 	@echo "****** compiling $(ONYVAIMAGC9B_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(ONYVAIMAGC9B_SRC) $(ONYVAIMAGC9B_INC) -o $@ 
 	@echo "## $(ONYVAIMAGC9B_CLAU) compiled successfully"
@@ -422,313 +262,115 @@
 
 # libk2vanderburg_fits
 #-----------------------------
-ifeq ($(ORDENA),pf-070210dl.intra.dlr.de)
-  K2VFTS_PATH = /scr2/cabr_ju/ross/blind/asalto27
-  K2VFTS_LIB_PATH = $(K2VFTS_PATH)/lib/$(ORDENA)
-  K2VFTS_INC_PATH = $(K2VFTS_PATH)/src
-else
-ifeq ($(ORDENA),OTS44)
-  K2VFTS_PATH = /data/COROT/qmon/asalto27
-  K2VFTS_LIB_PATH = $(K2VFTS_PATH)/lib/$(ORDENA)
-  K2VFTS_INC_PATH = $(K2VFTS_PATH)/src
-else
-ifeq ($(ORDENA),canijo)
-  K2VFTS_PATH = /datos/blind/asalto27
-  K2VFTS_LIB_PATH = $(K2VFTS_PATH)/lib/$(ORDENA)
-  K2VFTS_INC_PATH = $(K2VFTS_PATH)/src
-else
-ifeq ($(ORDENA),stahl)
-  K2VFTS_PATH = /home/juano/blind/asalto27
-  K2VFTS_LIB_PATH = $(K2VFTS_PATH)/lib/$(ORDENA)
-  K2VFTS_INC_PATH = $(K2VFTS_PATH)/src
-else
-ifeq ($(ORDENA),warg01)
-  K2VFTS_PATH = /home/warg01/cabrera/blind/asalto27
-  K2VFTS_LIB_PATH = $(K2VFTS_PATH)/lib/$(ORDENA)
-  K2VFTS_INC_PATH = $(K2VFTS_PATH)/src
-else
-ifeq ($(ORDENA),warg02)
-  K2VFTS_PATH = /home/warg01/cabrera/blind/asalto27
-  K2VFTS_LIB_PATH = $(K2VFTS_PATH)/lib/$(ORDENA)
-  K2VFTS_INC_PATH = $(K2VFTS_PATH)/src
-else
-  $(error error, computer $(ORDENA) not recognized)
-endif
-endif
-endif
-endif
-endif
-endif
+K2VFTS_PATH = ../asalto27
+K2VFTS_LIB_PATH = $(K2VFTS_PATH)/lib
+K2VFTS_INC_PATH = $(K2VFTS_PATH)/src
+
 K2VFTS_CLAU = k2vanderburg_fits
 K2VFTS_HDR = $(K2VFTS_PATH)/src/$(K2VFTS_CLAU).h
 K2VFTS_SRC = $(K2VFTS_PATH)/src/$(K2VFTS_CLAU).c
-K2VFTS_OBJ = $(K2VFTS_PATH)/obj/$(ORDENA)/$(K2VFTS_CLAU).o
-K2VFTS_LIB = $(K2VFTS_PATH)/lib/$(ORDENA)/lib$(K2VFTS_CLAU).a
+K2VFTS_OBJ = $(K2VFTS_PATH)/obj/$(K2VFTS_CLAU).o
+K2VFTS_LIB = $(K2VFTS_PATH)/lib/lib$(K2VFTS_CLAU).a
 K2VFTS_SLS = $(K2VFTS_SRC) $(K2VFTS_HDR) $(JUANFITS_HDR)
 K2VFTS_OLS = $(K2VFTS_OBJ)
 K2VFTS_LLS = $(K2VFTS_LIB) $(JUANFITS_LLS)
 K2VFTS_INC = -I$(K2VFTS_PATH)/src $(JUANFITS_INC) 
-K2VFTS_LNK = -L$(K2VFTS_PATH)/lib/$(ORDENA) -l$(K2VFTS_CLAU) $(JUANFITS_LNK) 
+K2VFTS_LNK = -L$(K2VFTS_PATH)/lib -l$(K2VFTS_CLAU) $(JUANFITS_LNK) 
 
 
 
 # libeverest_fits
 #-----------------------------
-ifeq ($(ORDENA),pf-070210dl.intra.dlr.de)
-  K2EFTS_PATH = /scr2/cabr_ju/ross/blind/asalto27
-  K2EFTS_LIB_PATH = $(K2EFTS_PATH)/lib/$(ORDENA)
-  K2EFTS_INC_PATH = $(K2EFTS_PATH)/src
-else
-ifeq ($(ORDENA),OTS44)
-  K2EFTS_PATH = /data/COROT/qmon/asalto27
-  K2EFTS_LIB_PATH = $(K2EFTS_PATH)/lib/$(ORDENA)
-  K2EFTS_INC_PATH = $(K2EFTS_PATH)/src
-else
-ifeq ($(ORDENA),canijo)
-  K2EFTS_PATH = /datos/blind/asalto27
-  K2EFTS_LIB_PATH = $(K2EFTS_PATH)/lib/$(ORDENA)
-  K2EFTS_INC_PATH = $(K2EFTS_PATH)/src
-else
-ifeq ($(ORDENA),stahl)
-  K2EFTS_PATH = /home/juano/blind/asalto27
-  K2EFTS_LIB_PATH = $(K2EFTS_PATH)/lib/$(ORDENA)
-  K2EFTS_INC_PATH = $(K2EFTS_PATH)/src
-else
-ifeq ($(ORDENA),warg01)
-  K2EFTS_PATH = /home/warg01/cabrera/blind/asalto27
-  K2EFTS_LIB_PATH = $(K2EFTS_PATH)/lib/$(ORDENA)
-  K2EFTS_INC_PATH = $(K2EFTS_PATH)/src
-else
-ifeq ($(ORDENA),warg02)
-  K2EFTS_PATH = /home/warg01/cabrera/blind/asalto27
-  K2EFTS_LIB_PATH = $(K2EFTS_PATH)/lib/$(ORDENA)
-  K2EFTS_INC_PATH = $(K2EFTS_PATH)/src
-else
-  $(error error, computer $(ORDENA) not recognized)
-endif
-endif
-endif
-endif
-endif
-endif
+K2EFTS_PATH = ../asalto27
+K2EFTS_LIB_PATH = $(K2EFTS_PATH)/lib
+K2EFTS_INC_PATH = $(K2EFTS_PATH)/src
+
 K2EFTS_CLAU = everest_fits
 K2EFTS_HDR = $(K2EFTS_PATH)/src/$(K2EFTS_CLAU).h
 K2EFTS_SRC = $(K2EFTS_PATH)/src/$(K2EFTS_CLAU).c
-K2EFTS_OBJ = $(K2EFTS_PATH)/obj/$(ORDENA)/$(K2EFTS_CLAU).o
-K2EFTS_LIB = $(K2EFTS_PATH)/lib/$(ORDENA)/lib$(K2EFTS_CLAU).a
+K2EFTS_OBJ = $(K2EFTS_PATH)/obj/$(K2EFTS_CLAU).o
+K2EFTS_LIB = $(K2EFTS_PATH)/lib/lib$(K2EFTS_CLAU).a
 K2EFTS_SLS = $(K2EFTS_SRC) $(K2EFTS_HDR) $(JUANFITS_HDR)
 K2EFTS_OLS = $(K2EFTS_OBJ)
 K2EFTS_LLS = $(K2EFTS_LIB) $(JUANFITS_LLS)
 K2EFTS_INC = -I$(K2EFTS_PATH)/src $(JUANFITS_INC) 
-K2EFTS_LNK = -L$(K2EFTS_PATH)/lib/$(ORDENA) -l$(K2EFTS_CLAU) $(JUANFITS_LNK) 
+K2EFTS_LNK = -L$(K2EFTS_PATH)/lib -l$(K2EFTS_CLAU) $(JUANFITS_LNK) 
 
 
 
 # libk2sc_fits
 #-----------------------------
-ifeq ($(ORDENA),pf-070210dl.intra.dlr.de)
-  K2SCFTS_PATH = /scr2/cabr_ju/ross/blind/asalto27
-  K2SCFTS_LIB_PATH = $(K2SCFTS_PATH)/lib/$(ORDENA)
-  K2SCFTS_INC_PATH = $(K2SCFTS_PATH)/src
-else
-ifeq ($(ORDENA),OTS44)
-  K2SCFTS_PATH = /data/COROT/qmon/asalto27
-  K2SCFTS_LIB_PATH = $(K2SCFTS_PATH)/lib/$(ORDENA)
-  K2SCFTS_INC_PATH = $(K2SCFTS_PATH)/src
-else
-ifeq ($(ORDENA),canijo)
-  K2SCFTS_PATH = /datos/blind/asalto27
-  K2SCFTS_LIB_PATH = $(K2SCFTS_PATH)/lib/$(ORDENA)
-  K2SCFTS_INC_PATH = $(K2SCFTS_PATH)/src
-else
-ifeq ($(ORDENA),stahl)
-  K2SCFTS_PATH = /home/juano/blind/asalto27
-  K2SCFTS_LIB_PATH = $(K2SCFTS_PATH)/lib/$(ORDENA)
-  K2SCFTS_INC_PATH = $(K2SCFTS_PATH)/src
-else
-ifeq ($(ORDENA),warg01)
-  K2SCFTS_PATH = /home/warg01/cabrera/blind/asalto27
-  K2SCFTS_LIB_PATH = $(K2SCFTS_PATH)/lib/$(ORDENA)
-  K2SCFTS_INC_PATH = $(K2SCFTS_PATH)/src
-else
-ifeq ($(ORDENA),warg02)
-  K2SCFTS_PATH = /home/warg01/cabrera/blind/asalto27
-  K2SCFTS_LIB_PATH = $(K2SCFTS_PATH)/lib/$(ORDENA)
-  K2SCFTS_INC_PATH = $(K2SCFTS_PATH)/src
-else
-  $(error error, computer $(ORDENA) not recognized)
-endif
-endif
-endif
-endif
-endif
-endif
+K2SCFTS_PATH = ../asalto27
+K2SCFTS_LIB_PATH = $(K2SCFTS_PATH)/lib
+K2SCFTS_INC_PATH = $(K2SCFTS_PATH)/src
+
 K2SCFTS_CLAU = k2sc_fits
 K2SCFTS_HDR = $(K2SCFTS_PATH)/src/$(K2SCFTS_CLAU).h
 K2SCFTS_SRC = $(K2SCFTS_PATH)/src/$(K2SCFTS_CLAU).c
-K2SCFTS_OBJ = $(K2SCFTS_PATH)/obj/$(ORDENA)/$(K2SCFTS_CLAU).o
-K2SCFTS_LIB = $(K2SCFTS_PATH)/lib/$(ORDENA)/lib$(K2SCFTS_CLAU).a
+K2SCFTS_OBJ = $(K2SCFTS_PATH)/obj/$(K2SCFTS_CLAU).o
+K2SCFTS_LIB = $(K2SCFTS_PATH)/lib/lib$(K2SCFTS_CLAU).a
 K2SCFTS_SLS = $(K2SCFTS_SRC) $(K2SCFTS_HDR) $(JUANFITS_HDR)
 K2SCFTS_OLS = $(K2SCFTS_OBJ)
 K2SCFTS_LLS = $(K2SCFTS_LIB) $(JUANFITS_LLS)
 K2SCFTS_INC = -I$(K2SCFTS_PATH)/src $(JUANFITS_INC) 
-K2SCFTS_LNK = -L$(K2SCFTS_PATH)/lib/$(ORDENA) -l$(K2SCFTS_CLAU) $(JUANFITS_LNK) 
+K2SCFTS_LNK = -L$(K2SCFTS_PATH)/lib -l$(K2SCFTS_CLAU) $(JUANFITS_LNK) 
 
 
 
 # libpolar_fits
 #-----------------------------
-ifeq ($(ORDENA),pf-070210dl.intra.dlr.de)
-  K2PFTS_PATH = /scr2/cabr_ju/ross/blind/asalto27
-  K2PFTS_LIB_PATH = $(K2PFTS_PATH)/lib/$(ORDENA)
-  K2PFTS_INC_PATH = $(K2PFTS_PATH)/src
-else
-ifeq ($(ORDENA),OTS44)
-  K2PFTS_PATH = /data/COROT/qmon/asalto27
-  K2PFTS_LIB_PATH = $(K2PFTS_PATH)/lib/$(ORDENA)
-  K2PFTS_INC_PATH = $(K2PFTS_PATH)/src
-else
-ifeq ($(ORDENA),canijo)
-  K2PFTS_PATH = /datos/blind/asalto27
-  K2PFTS_LIB_PATH = $(K2PFTS_PATH)/lib/$(ORDENA)
-  K2PFTS_INC_PATH = $(K2PFTS_PATH)/src
-else
-ifeq ($(ORDENA),stahl)
-  K2PFTS_PATH = /home/juano/blind/asalto27
-  K2PFTS_LIB_PATH = $(K2PFTS_PATH)/lib/$(ORDENA)
-  K2PFTS_INC_PATH = $(K2PFTS_PATH)/src
-else
-ifeq ($(ORDENA),warg01)
-  K2PFTS_PATH = /home/warg01/cabrera/blind/asalto27
-  K2PFTS_LIB_PATH = $(K2PFTS_PATH)/lib/$(ORDENA)
-  K2PFTS_INC_PATH = $(K2PFTS_PATH)/src
-else
-ifeq ($(ORDENA),warg02)
-  K2PFTS_PATH = /home/warg01/cabrera/blind/asalto27
-  K2PFTS_LIB_PATH = $(K2PFTS_PATH)/lib/$(ORDENA)
-  K2PFTS_INC_PATH = $(K2PFTS_PATH)/src
-else
-  $(error error, computer $(ORDENA) not recognized)
-endif
-endif
-endif
-endif
-endif
-endif
+K2PFTS_PATH = ../asalto27
+K2PFTS_LIB_PATH = $(K2PFTS_PATH)/lib
+K2PFTS_INC_PATH = $(K2PFTS_PATH)/src
+
 K2PFTS_CLAU = polar_fits
 K2PFTS_HDR = $(K2PFTS_PATH)/src/$(K2PFTS_CLAU).h
 K2PFTS_SRC = $(K2PFTS_PATH)/src/$(K2PFTS_CLAU).c
-K2PFTS_OBJ = $(K2PFTS_PATH)/obj/$(ORDENA)/$(K2PFTS_CLAU).o
-K2PFTS_LIB = $(K2PFTS_PATH)/lib/$(ORDENA)/lib$(K2PFTS_CLAU).a
+K2PFTS_OBJ = $(K2PFTS_PATH)/obj/$(K2PFTS_CLAU).o
+K2PFTS_LIB = $(K2PFTS_PATH)/lib/lib$(K2PFTS_CLAU).a
 K2PFTS_SLS = $(K2PFTS_SRC) $(K2PFTS_HDR) $(JUANFITS_HDR)
 K2PFTS_OLS = $(K2PFTS_OBJ)
 K2PFTS_LLS = $(K2PFTS_LIB) $(JUANFITS_LLS)
 K2PFTS_INC = -I$(K2PFTS_PATH)/src $(JUANFITS_INC) 
-K2PFTS_LNK = -L$(K2PFTS_PATH)/lib/$(ORDENA) -l$(K2PFTS_CLAU) $(JUANFITS_LNK) 
+K2PFTS_LNK = -L$(K2PFTS_PATH)/lib -l$(K2PFTS_CLAU) $(JUANFITS_LNK) 
 
 
 
 # libkepler_fits
 #-----------------------------
-ifeq ($(ORDENA),pf-070210dl.intra.dlr.de)
-  KFTS_PATH = /scr2/cabr_ju/ross/blind/asalto27
-  KFTS_LIB_PATH = $(KFTS_PATH)/lib/$(ORDENA)
-  KFTS_INC_PATH = $(KFTS_PATH)/src
-else
-ifeq ($(ORDENA),OTS44)
-  KFTS_PATH = /data/COROT/qmon/asalto27
-  KFTS_LIB_PATH = $(KFTS_PATH)/lib/$(ORDENA)
-  KFTS_INC_PATH = $(KFTS_PATH)/src
-else
-ifeq ($(ORDENA),canijo)
-  KFTS_PATH = /datos/blind/asalto27
-  KFTS_LIB_PATH = $(KFTS_PATH)/lib/$(ORDENA)
-  KFTS_INC_PATH = $(KFTS_PATH)/src
-else
-ifeq ($(ORDENA),stahl)
-  KFTS_PATH = /home/juano/blind/asalto27
-  KFTS_LIB_PATH = $(KFTS_PATH)/lib/$(ORDENA)
-  KFTS_INC_PATH = $(KFTS_PATH)/src
-else
-ifeq ($(ORDENA),warg01)
-  KFTS_PATH = /home/warg01/cabrera/blind/asalto27
-  KFTS_LIB_PATH = $(KFTS_PATH)/lib/$(ORDENA)
-  KFTS_INC_PATH = $(KFTS_PATH)/src
-else
-ifeq ($(ORDENA),warg02)
-  KFTS_PATH = /home/warg01/cabrera/blind/asalto27
-  KFTS_LIB_PATH = $(KFTS_PATH)/lib/$(ORDENA)
-  KFTS_INC_PATH = $(KFTS_PATH)/src
-else
-  $(error error, computer $(ORDENA) not recognized)
-endif
-endif
-endif
-endif
-endif
-endif
+KFTS_PATH = ../asalto27
+KFTS_LIB_PATH = $(KFTS_PATH)/lib
+KFTS_INC_PATH = $(KFTS_PATH)/src
+
 KFTS_CLAU = kepler_fits
 KFTS_HDR = $(KFTS_PATH)/src/$(KFTS_CLAU).h
 KFTS_SRC = $(KFTS_PATH)/src/$(KFTS_CLAU).c
-KFTS_OBJ = $(KFTS_PATH)/obj/$(ORDENA)/$(KFTS_CLAU).o
-KFTS_LIB = $(KFTS_PATH)/lib/$(ORDENA)/lib$(KFTS_CLAU).a
+KFTS_OBJ = $(KFTS_PATH)/obj/$(KFTS_CLAU).o
+KFTS_LIB = $(KFTS_PATH)/lib/lib$(KFTS_CLAU).a
 KFTS_SLS = $(KFTS_SRC) $(KFTS_HDR) $(JUANFITS_HDR)
 KFTS_OLS = $(KFTS_OBJ)
 KFTS_LLS = $(KFTS_LIB) $(JUANFITS_LLS)
 KFTS_INC = -I$(KFTS_PATH)/src $(JUANFITS_INC) 
-KFTS_LNK = -L$(KFTS_PATH)/lib/$(ORDENA) -l$(KFTS_CLAU) $(JUANFITS_LNK) 
+KFTS_LNK = -L$(KFTS_PATH)/lib -l$(KFTS_CLAU) $(JUANFITS_LNK) 
 
 
 
 # libtess_fits
 #-----------------------------
-ifeq ($(ORDENA),pf-070210dl.intra.dlr.de)
-  TFTS_PATH = /scr2/cabr_ju/ross/blind/asalto27
-  TFTS_LIB_PATH = $(TFTS_PATH)/lib/$(ORDENA)
-  TFTS_INC_PATH = $(TFTS_PATH)/src
-else
-ifeq ($(ORDENA),OTS44)
-  TFTS_PATH = /data/COROT/qmon/asalto27
-  TFTS_LIB_PATH = $(TFTS_PATH)/lib/$(ORDENA)
-  TFTS_INC_PATH = $(TFTS_PATH)/src
-else
-ifeq ($(ORDENA),canijo)
-  TFTS_PATH = /datos/blind/asalto27
-  TFTS_LIB_PATH = $(TFTS_PATH)/lib/$(ORDENA)
-  TFTS_INC_PATH = $(TFTS_PATH)/src
-else
-ifeq ($(ORDENA),stahl)
-  TFTS_PATH = /home/juano/blind/asalto27
-  TFTS_LIB_PATH = $(TFTS_PATH)/lib/$(ORDENA)
-  TFTS_INC_PATH = $(TFTS_PATH)/src
-else
-ifeq ($(ORDENA),warg01)
-  TFTS_PATH = /home/warg01/cabrera/blind/asalto27
-  TFTS_LIB_PATH = $(TFTS_PATH)/lib/$(ORDENA)
-  TFTS_INC_PATH = $(TFTS_PATH)/src
-else
-ifeq ($(ORDENA),warg02)
-  TFTS_PATH = /home/warg01/cabrera/blind/asalto27
-  TFTS_LIB_PATH = $(TFTS_PATH)/lib/$(ORDENA)
-  TFTS_INC_PATH = $(TFTS_PATH)/src
-else
-  $(error error, computer $(ORDENA) not recognized)
-endif
-endif
-endif
-endif
-endif
-endif
+TFTS_PATH = ../asalto27
+TFTS_LIB_PATH = $(TFTS_PATH)/lib
+TFTS_INC_PATH = $(TFTS_PATH)/src
+
 TFTS_CLAU = tess_fits
 TFTS_HDR = $(TFTS_PATH)/src/$(TFTS_CLAU).h
 TFTS_SRC = $(TFTS_PATH)/src/$(TFTS_CLAU).c
-TFTS_OBJ = $(TFTS_PATH)/obj/$(ORDENA)/$(TFTS_CLAU).o
-TFTS_LIB = $(TFTS_PATH)/lib/$(ORDENA)/lib$(TFTS_CLAU).a
+TFTS_OBJ = $(TFTS_PATH)/obj/$(TFTS_CLAU).o
+TFTS_LIB = $(TFTS_PATH)/lib/lib$(TFTS_CLAU).a
 TFTS_SLS = $(TFTS_SRC) $(TFTS_HDR) $(JUANFITS_HDR)
 TFTS_OLS = $(TFTS_OBJ)
 TFTS_LLS = $(TFTS_LIB) $(JUANFITS_LLS)
 TFTS_INC = -I$(TFTS_PATH)/src $(JUANFITS_INC) 
-TFTS_LNK = -L$(TFTS_PATH)/lib/$(ORDENA) -l$(TFTS_CLAU) $(JUANFITS_LNK) 
+TFTS_LNK = -L$(TFTS_PATH)/lib -l$(TFTS_CLAU) $(JUANFITS_LNK) 
 
 
 
@@ -736,8 +378,8 @@
 #-----------------------------
 ONYVAKEPLER_CLAU = onyva_kepler
 ONYVAKEPLER_SRC = $(MAIN_PATH)/src/$(ONYVAKEPLER_CLAU).c
-ONYVAKEPLER_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(ONYVAKEPLER_CLAU).o
-ONYVAKEPLER_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(ONYVAKEPLER_CLAU).exe
+ONYVAKEPLER_OBJ = $(MAIN_PATH)/obj/$(ONYVAKEPLER_CLAU).o
+ONYVAKEPLER_EXE = $(MAIN_PATH)/bin/$(ONYVAKEPLER_CLAU).exe
 ONYVAKEPLER_SLS = $(ONYVAKEPLER_SRC) $(ONYVAKEPLER_HDR)
 ONYVAKEPLER_OLS = $(ONYVAKEPLER_OBJ)
 ONYVAKEPLER_LLS = $(BT2_LLS) $(KFTS_LLS)
@@ -747,12 +389,12 @@
 $(ONYVAKEPLER_CLAU) : $(ONYVAKEPLER_EXE)
 	@echo " clau: $(ONYVAKEPLER_CLAU)"
 
-$(ONYVAKEPLER_EXE) : $(ONYVAKEPLER_OLS) $(ONYVAKEPLER_LLS)
+$(ONYVAKEPLER_EXE) : $(ONYVAKEPLER_OLS) $(ONYVAKEPLER_LLS) make_directories
 	@echo "****** linking $(ONYVAKEPLER_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(ONYVAKEPLER_OLS) -o $@ $(ONYVAKEPLER_LNK) 
+	$(COMPILADOR) $(FLAGS) $(ONYVAKEPLER_OLS) -o $@ $(ONYVAKEPLER_LNK) $(JUANFITS_OBJ)
 	@echo "## $(ONYVAKEPLER_CLAU) linked successfully"
 
-$(ONYVAKEPLER_OBJ) : $(ONYVAKEPLER_SLS)
+$(ONYVAKEPLER_OBJ) : $(ONYVAKEPLER_SLS) make_directories
 	@echo "****** compiling $(ONYVAKEPLER_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(ONYVAKEPLER_SRC) $(ONYVAKEPLER_INC) -o $@ 
 	@echo "## $(ONYVAKEPLER_CLAU) compiled successfully"
@@ -763,8 +405,8 @@
 #-----------------------------
 ONYVAKEPLEREDUARDO_CLAU = onyva_kepler_eduardo
 ONYVAKEPLEREDUARDO_SRC = $(MAIN_PATH)/src/$(ONYVAKEPLEREDUARDO_CLAU).c
-ONYVAKEPLEREDUARDO_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(ONYVAKEPLEREDUARDO_CLAU).o
-ONYVAKEPLEREDUARDO_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(ONYVAKEPLEREDUARDO_CLAU).exe
+ONYVAKEPLEREDUARDO_OBJ = $(MAIN_PATH)/obj/$(ONYVAKEPLEREDUARDO_CLAU).o
+ONYVAKEPLEREDUARDO_EXE = $(MAIN_PATH)/bin/$(ONYVAKEPLEREDUARDO_CLAU).exe
 ONYVAKEPLEREDUARDO_SLS = $(ONYVAKEPLEREDUARDO_SRC) $(ONYVAKEPLEREDUARDO_HDR)
 ONYVAKEPLEREDUARDO_OLS = $(ONYVAKEPLEREDUARDO_OBJ)
 ONYVAKEPLEREDUARDO_LLS = $(BT2_LLS) $(KFTS_LLS)
@@ -774,12 +416,12 @@
 $(ONYVAKEPLEREDUARDO_CLAU) : $(ONYVAKEPLEREDUARDO_EXE)
 	@echo " clau: $(ONYVAKEPLEREDUARDO_CLAU)"
 
-$(ONYVAKEPLEREDUARDO_EXE) : $(ONYVAKEPLEREDUARDO_OLS) $(ONYVAKEPLEREDUARDO_LLS)
+$(ONYVAKEPLEREDUARDO_EXE) : $(ONYVAKEPLEREDUARDO_OLS) $(ONYVAKEPLEREDUARDO_LLS) make_directories
 	@echo "****** linking $(ONYVAKEPLEREDUARDO_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(ONYVAKEPLEREDUARDO_OLS) -o $@ $(ONYVAKEPLEREDUARDO_LNK) 
+	$(COMPILADOR) $(FLAGS) $(ONYVAKEPLEREDUARDO_OLS) -o $@ $(ONYVAKEPLEREDUARDO_LNK) $(JUANFITS_OBJ) 
 	@echo "## $(ONYVAKEPLEREDUARDO_CLAU) linked successfully"
 
-$(ONYVAKEPLEREDUARDO_OBJ) : $(ONYVAKEPLEREDUARDO_SLS)
+$(ONYVAKEPLEREDUARDO_OBJ) : $(ONYVAKEPLEREDUARDO_SLS) make_directories
 	@echo "****** compiling $(ONYVAKEPLEREDUARDO_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(ONYVAKEPLEREDUARDO_SRC) $(ONYVAKEPLEREDUARDO_INC) -o $@ 
 	@echo "## $(ONYVAKEPLEREDUARDO_CLAU) compiled successfully"
@@ -790,8 +432,8 @@
 #-----------------------------
 ONYVAK2VANDERBURG_CLAU = onyva_k2vanderburg
 ONYVAK2VANDERBURG_SRC = $(MAIN_PATH)/src/$(ONYVAK2VANDERBURG_CLAU).c
-ONYVAK2VANDERBURG_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(ONYVAK2VANDERBURG_CLAU).o
-ONYVAK2VANDERBURG_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(ONYVAK2VANDERBURG_CLAU).exe
+ONYVAK2VANDERBURG_OBJ = $(MAIN_PATH)/obj/$(ONYVAK2VANDERBURG_CLAU).o
+ONYVAK2VANDERBURG_EXE = $(MAIN_PATH)/bin/$(ONYVAK2VANDERBURG_CLAU).exe
 ONYVAK2VANDERBURG_SLS = $(ONYVAK2VANDERBURG_SRC) $(ONYVAK2VANDERBURG_HDR) $(K2VFTS_HDR)
 ONYVAK2VANDERBURG_OLS = $(ONYVAK2VANDERBURG_OBJ)
 ONYVAK2VANDERBURG_LLS = $(BT2_LLS) $(K2VFTS_LLS)
@@ -801,14 +443,14 @@
 $(ONYVAK2VANDERBURG_CLAU) : $(ONYVAK2VANDERBURG_EXE)
 	@echo " clau: $(ONYVAK2VANDERBURG_CLAU)"
 
-$(ONYVAK2VANDERBURG_EXE) : $(ONYVAK2VANDERBURG_OLS) $(ONYVAK2VANDERBURG_LLS)
+$(ONYVAK2VANDERBURG_EXE) : $(ONYVAK2VANDERBURG_OLS) $(ONYVAK2VANDERBURG_LLS) make_directories
 	@echo "****** linking $(ONYVAK2VANDERBURG_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(ONYVAK2VANDERBURG_OLS) -o $@ $(ONYVAK2VANDERBURG_LNK) 
+	$(COMPILADOR) $(FLAGS) $(ONYVAK2VANDERBURG_OLS) -o $@ $(ONYVAK2VANDERBURG_LNK) $(JUANFITS_OBJ)
 	@echo "## $(ONYVAK2VANDERBURG_CLAU) linked successfully"
 
-$(ONYVAK2VANDERBURG_OBJ) : $(ONYVAK2VANDERBURG_SLS)
+$(ONYVAK2VANDERBURG_OBJ) : $(ONYVAK2VANDERBURG_SLS) make_directories
 	@echo "****** compiling $(ONYVAK2VANDERBURG_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) -c $(ONYVAK2VANDERBURG_SRC) $(ONYVAK2VANDERBURG_INC) -o $@ 
+	$(COMPILADOR) $(FLAGS) -c $(ONYVAK2VANDERBURG_SRC) $(ONYVAK2VANDERBURG_INC) -o $@ -I$(JUANFITS_PATH) 
 	@echo "## $(ONYVAK2VANDERBURG_CLAU) compiled successfully"
 
 
@@ -817,8 +459,8 @@
 #-----------------------------
 ONYVAEVEREST_CLAU = onyva_everest
 ONYVAEVEREST_SRC = $(MAIN_PATH)/src/$(ONYVAEVEREST_CLAU).c
-ONYVAEVEREST_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(ONYVAEVEREST_CLAU).o
-ONYVAEVEREST_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(ONYVAEVEREST_CLAU).exe
+ONYVAEVEREST_OBJ = $(MAIN_PATH)/obj/$(ONYVAEVEREST_CLAU).o
+ONYVAEVEREST_EXE = $(MAIN_PATH)/bin/$(ONYVAEVEREST_CLAU).exe
 ONYVAEVEREST_SLS = $(ONYVAEVEREST_SRC) $(ONYVAEVEREST_HDR) $(K2EFTS_HDR)
 ONYVAEVEREST_OLS = $(ONYVAEVEREST_OBJ)
 ONYVAEVEREST_LLS = $(BT2_LLS) $(K2EFTS_LLS)
@@ -828,14 +470,14 @@
 $(ONYVAEVEREST_CLAU) : $(ONYVAEVEREST_EXE)
 	@echo " clau: $(ONYVAEVEREST_CLAU)"
 
-$(ONYVAEVEREST_EXE) : $(ONYVAEVEREST_OLS) $(ONYVAEVEREST_LLS)
+$(ONYVAEVEREST_EXE) : $(ONYVAEVEREST_OLS) $(ONYVAEVEREST_LLS) make_directories
 	@echo "****** linking $(ONYVAEVEREST_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(ONYVAEVEREST_OLS) -o $@ $(ONYVAEVEREST_LNK) 
+	$(COMPILADOR) $(FLAGS) $(ONYVAEVEREST_OLS) -o $@ $(ONYVAEVEREST_LNK) $(JUANFITS_OBJ)
 	@echo "## $(ONYVAEVEREST_CLAU) linked successfully"
 
-$(ONYVAEVEREST_OBJ) : $(ONYVAEVEREST_SLS)
+$(ONYVAEVEREST_OBJ) : $(ONYVAEVEREST_SLS) make_directories
 	@echo "****** compiling $(ONYVAEVEREST_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) -c $(ONYVAEVEREST_SRC) $(ONYVAEVEREST_INC) -o $@ 
+	$(COMPILADOR) $(FLAGS) -c $(ONYVAEVEREST_SRC) $(ONYVAEVEREST_INC) -o $@ -I$(JUANFITS_PATH)
 	@echo "## $(ONYVAEVEREST_CLAU) compiled successfully"
 
 
@@ -844,8 +486,8 @@
 #-----------------------------
 ONYVAK2SC_CLAU = onyva_k2sc
 ONYVAK2SC_SRC = $(MAIN_PATH)/src/$(ONYVAK2SC_CLAU).c
-ONYVAK2SC_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(ONYVAK2SC_CLAU).o
-ONYVAK2SC_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(ONYVAK2SC_CLAU).exe
+ONYVAK2SC_OBJ = $(MAIN_PATH)/obj/$(ONYVAK2SC_CLAU).o
+ONYVAK2SC_EXE = $(MAIN_PATH)/bin/$(ONYVAK2SC_CLAU).exe
 ONYVAK2SC_SLS = $(ONYVAK2SC_SRC) $(ONYVAK2SC_HDR) $(K2SCFTS_HDR)
 ONYVAK2SC_OLS = $(ONYVAK2SC_OBJ)
 ONYVAK2SC_LLS = $(BT2_LLS) $(K2SCFTS_LLS)
@@ -855,12 +497,12 @@
 $(ONYVAK2SC_CLAU) : $(ONYVAK2SC_EXE)
 	@echo " clau: $(ONYVAK2SC_CLAU)"
 
-$(ONYVAK2SC_EXE) : $(ONYVAK2SC_OLS) $(ONYVAK2SC_LLS)
+$(ONYVAK2SC_EXE) : $(ONYVAK2SC_OLS) $(ONYVAK2SC_LLS) make_directories
 	@echo "****** linking $(ONYVAK2SC_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(ONYVAK2SC_OLS) -o $@ $(ONYVAK2SC_LNK) 
+	$(COMPILADOR) $(FLAGS) $(ONYVAK2SC_OLS) -o $@ $(ONYVAK2SC_LNK) $(JUANFITS_OBJ)
 	@echo "## $(ONYVAK2SC_CLAU) linked successfully"
 
-$(ONYVAK2SC_OBJ) : $(ONYVAK2SC_SLS)
+$(ONYVAK2SC_OBJ) : $(ONYVAK2SC_SLS) make_directories
 	@echo "****** compiling $(ONYVAK2SC_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(ONYVAK2SC_SRC) $(ONYVAK2SC_INC) -o $@ 
 	@echo "## $(ONYVAK2SC_CLAU) compiled successfully"
@@ -871,8 +513,8 @@
 #-----------------------------
 ONYVAPOLAR_CLAU = onyva_polar
 ONYVAPOLAR_SRC = $(MAIN_PATH)/src/$(ONYVAPOLAR_CLAU).c
-ONYVAPOLAR_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(ONYVAPOLAR_CLAU).o
-ONYVAPOLAR_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(ONYVAPOLAR_CLAU).exe
+ONYVAPOLAR_OBJ = $(MAIN_PATH)/obj/$(ONYVAPOLAR_CLAU).o
+ONYVAPOLAR_EXE = $(MAIN_PATH)/bin/$(ONYVAPOLAR_CLAU).exe
 ONYVAPOLAR_SLS = $(ONYVAPOLAR_SRC) $(ONYVAPOLAR_HDR) $(K2PFTS_HDR)
 ONYVAPOLAR_OLS = $(ONYVAPOLAR_OBJ)
 ONYVAPOLAR_LLS = $(BT2_LLS) $(K2PFTS_LLS)
@@ -882,12 +524,12 @@
 $(ONYVAPOLAR_CLAU) : $(ONYVAPOLAR_EXE)
 	@echo " clau: $(ONYVAPOLAR_CLAU)"
 
-$(ONYVAPOLAR_EXE) : $(ONYVAPOLAR_OLS) $(ONYVAPOLAR_LLS)
+$(ONYVAPOLAR_EXE) : $(ONYVAPOLAR_OLS) $(ONYVAPOLAR_LLS) make_directories
 	@echo "****** linking $(ONYVAPOLAR_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(ONYVAPOLAR_OLS) -o $@ $(ONYVAPOLAR_LNK) 
+	$(COMPILADOR) $(FLAGS) $(ONYVAPOLAR_OLS) -o $@ $(ONYVAPOLAR_LNK) $(JUANFITS_OBJ)
 	@echo "## $(ONYVAPOLAR_CLAU) linked successfully"
 
-$(ONYVAPOLAR_OBJ) : $(ONYVAPOLAR_SLS)
+$(ONYVAPOLAR_OBJ) : $(ONYVAPOLAR_SLS) make_directories
 	@echo "****** compiling $(ONYVAPOLAR_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(ONYVAPOLAR_SRC) $(ONYVAPOLAR_INC) -o $@ 
 	@echo "## $(ONYVAPOLAR_CLAU) compiled successfully"
@@ -898,8 +540,8 @@
 #-----------------------------
 ONYVATESS_CLAU = onyva_tess
 ONYVATESS_SRC = $(MAIN_PATH)/src/$(ONYVATESS_CLAU).c
-ONYVATESS_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(ONYVATESS_CLAU).o
-ONYVATESS_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(ONYVATESS_CLAU).exe
+ONYVATESS_OBJ = $(MAIN_PATH)/obj/$(ONYVATESS_CLAU).o
+ONYVATESS_EXE = $(MAIN_PATH)/bin/$(ONYVATESS_CLAU).exe
 ONYVATESS_SLS = $(ONYVATESS_SRC) $(ONYVATESS_HDR)
 ONYVATESS_OLS = $(ONYVATESS_OBJ)
 ONYVATESS_LLS = $(BT2_LLS) $(TFTS_LLS)
@@ -909,12 +551,12 @@
 $(ONYVATESS_CLAU) : $(ONYVATESS_EXE)
 	@echo " clau: $(ONYVATESS_CLAU)"
 
-$(ONYVATESS_EXE) : $(ONYVATESS_OLS) $(ONYVATESS_LLS)
+$(ONYVATESS_EXE) : $(ONYVATESS_OLS) $(ONYVATESS_LLS) make_directories
 	@echo "****** linking $(ONYVATESS_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(ONYVATESS_OLS) -o $@ $(ONYVATESS_LNK) 
+	$(COMPILADOR) $(FLAGS) $(ONYVATESS_OLS) -o $@ $(ONYVATESS_LNK) $(JUANFITS_OBJ)
 	@echo "## $(ONYVATESS_CLAU) linked successfully"
 
-$(ONYVATESS_OBJ) : $(ONYVATESS_SLS)
+$(ONYVATESS_OBJ) : $(ONYVATESS_SLS) make_directories
 	@echo "****** compiling $(ONYVATESS_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(ONYVATESS_SRC) $(ONYVATESS_INC) -o $@ 
 	@echo "## $(ONYVATESS_CLAU) compiled successfully"
@@ -925,8 +567,8 @@
 #-----------------------------
 ONYVACHEOPS_CLAU = onyva_cheops
 ONYVACHEOPS_SRC = $(MAIN_PATH)/src/$(ONYVACHEOPS_CLAU).c
-ONYVACHEOPS_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(ONYVACHEOPS_CLAU).o
-ONYVACHEOPS_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(ONYVACHEOPS_CLAU).exe
+ONYVACHEOPS_OBJ = $(MAIN_PATH)/obj/$(ONYVACHEOPS_CLAU).o
+ONYVACHEOPS_EXE = $(MAIN_PATH)/bin/$(ONYVACHEOPS_CLAU).exe
 ONYVACHEOPS_SLS = $(ONYVACHEOPS_SRC) $(ONYVACHEOPS_HDR)
 ONYVACHEOPS_OLS = $(ONYVACHEOPS_OBJ)
 ONYVACHEOPS_LLS = $(BT2_LLS) $(TFTS_LLS)
@@ -936,12 +578,12 @@
 $(ONYVACHEOPS_CLAU) : $(ONYVACHEOPS_EXE)
 	@echo " clau: $(ONYVACHEOPS_CLAU)"
 
-$(ONYVACHEOPS_EXE) : $(ONYVACHEOPS_OLS) $(ONYVACHEOPS_LLS)
+$(ONYVACHEOPS_EXE) : $(ONYVACHEOPS_OLS) $(ONYVACHEOPS_LLS) make_directories
 	@echo "****** linking $(ONYVACHEOPS_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(ONYVACHEOPS_OLS) -o $@ $(ONYVACHEOPS_LNK) 
+	$(COMPILADOR) $(FLAGS) $(ONYVACHEOPS_OLS) -o $@ $(ONYVACHEOPS_LNK) $(JUANFITS_OBJ)
 	@echo "## $(ONYVACHEOPS_CLAU) linked successfully"
 
-$(ONYVACHEOPS_OBJ) : $(ONYVACHEOPS_SLS)
+$(ONYVACHEOPS_OBJ) : $(ONYVACHEOPS_SLS) make_directories
 	@echo "****** compiling $(ONYVACHEOPS_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(ONYVACHEOPS_SRC) $(ONYVACHEOPS_INC) -o $@ 
 	@echo "## $(ONYVACHEOPS_CLAU) compiled successfully"
@@ -952,8 +594,8 @@
 #-----------------------------
 ONYVATASOC_CLAU = onyva_tasoc
 ONYVATASOC_SRC = $(MAIN_PATH)/src/$(ONYVATASOC_CLAU).c
-ONYVATASOC_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(ONYVATASOC_CLAU).o
-ONYVATASOC_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(ONYVATASOC_CLAU).exe
+ONYVATASOC_OBJ = $(MAIN_PATH)/obj/$(ONYVATASOC_CLAU).o
+ONYVATASOC_EXE = $(MAIN_PATH)/bin/$(ONYVATASOC_CLAU).exe
 ONYVATASOC_SLS = $(ONYVATASOC_SRC) $(ONYVATASOC_HDR)
 ONYVATASOC_OLS = $(ONYVATASOC_OBJ)
 ONYVATASOC_LLS = $(BT2_LLS) $(TFTS_LLS)
@@ -963,12 +605,12 @@
 $(ONYVATASOC_CLAU) : $(ONYVATASOC_EXE)
 	@echo " clau: $(ONYVATASOC_CLAU)"
 
-$(ONYVATASOC_EXE) : $(ONYVATASOC_OLS) $(ONYVATASOC_LLS)
+$(ONYVATASOC_EXE) : $(ONYVATASOC_OLS) $(ONYVATASOC_LLS) make_directories
 	@echo "****** linking $(ONYVATASOC_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(ONYVATASOC_OLS) -o $@ $(ONYVATASOC_LNK) 
+	$(COMPILADOR) $(FLAGS) $(ONYVATASOC_OLS) -o $@ $(ONYVATASOC_LNK) $(JUANFITS_OBJ)
 	@echo "## $(ONYVATASOC_CLAU) linked successfully"
 
-$(ONYVATASOC_OBJ) : $(ONYVATASOC_SLS)
+$(ONYVATASOC_OBJ) : $(ONYVATASOC_SLS) make_directories
 	@echo "****** compiling $(ONYVATASOC_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(ONYVATASOC_SRC) $(ONYVATASOC_INC) -o $@ 
 	@echo "## $(ONYVATASOC_CLAU) compiled successfully"
@@ -979,8 +621,8 @@
 #-----------------------------
 ONYVACLNSET_CLAU = onyva_clnset
 ONYVACLNSET_SRC = $(MAIN_PATH)/src/$(ONYVACLNSET_CLAU).c
-ONYVACLNSET_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(ONYVACLNSET_CLAU).o
-ONYVACLNSET_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(ONYVACLNSET_CLAU).exe
+ONYVACLNSET_OBJ = $(MAIN_PATH)/obj/$(ONYVACLNSET_CLAU).o
+ONYVACLNSET_EXE = $(MAIN_PATH)/bin/$(ONYVACLNSET_CLAU).exe
 ONYVACLNSET_SLS = $(ONYVACLNSET_SRC) $(BT2_HDR)
 ONYVACLNSET_OLS = $(ONYVACLNSET_OBJ) 
 ONYVACLNSET_LLS = $(BT2_LLS) 
@@ -990,12 +632,12 @@
 $(ONYVACLNSET_CLAU) : $(ONYVACLNSET_EXE)
 	@echo " clau: $(ONYVACLNSET_CLAU)"
 
-$(ONYVACLNSET_EXE) : $(ONYVACLNSET_OLS) $(ONYVACLNSET_LLS)
+$(ONYVACLNSET_EXE) : $(ONYVACLNSET_OLS) $(ONYVACLNSET_LLS) make_directories
 	@echo "****** linking $(ONYVACLNSET_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(ONYVACLNSET_OLS) -o $@ $(ONYVACLNSET_LNK) 
+	$(COMPILADOR) $(FLAGS) $(ONYVACLNSET_OLS) -o $@ $(ONYVACLNSET_LNK) $(JUANFITS_OBJ)
 	@echo "## $(ONYVACLNSET_CLAU) linked successfully"
 
-$(ONYVACLNSET_OBJ) : $(ONYVACLNSET_SLS)
+$(ONYVACLNSET_OBJ) : $(ONYVACLNSET_SLS) make_directories
 	@echo "****** compiling $(ONYVACLNSET_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(ONYVACLNSET_SRC) $(ONYVACLNSET_INC) -o $@ 
 	@echo "## $(ONYVACLNSET_CLAU) compiled successfully"
@@ -1006,8 +648,8 @@
 #-----------------------------
 TSEVI_CLAU = tsevi
 TSEVI_SRC = $(MAIN_PATH)/src/$(TSEVI_CLAU).c
-TSEVI_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(TSEVI_CLAU).o
-TSEVI_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(TSEVI_CLAU).exe
+TSEVI_OBJ = $(MAIN_PATH)/obj/$(TSEVI_CLAU).o
+TSEVI_EXE = $(MAIN_PATH)/bin/$(TSEVI_CLAU).exe
 TSEVI_SLS = $(TSEVI_SRC) $(BT2_HDR)
 TSEVI_OLS = $(TSEVI_OBJ) 
 TSEVI_LLS = $(BT2_LLS) 
@@ -1017,12 +659,12 @@
 $(TSEVI_CLAU) : $(TSEVI_EXE)
 	@echo " clau: $(TSEVI_CLAU)"
 
-$(TSEVI_EXE) : $(TSEVI_OLS) $(TSEVI_LLS)
+$(TSEVI_EXE) : $(TSEVI_OLS) $(TSEVI_LLS) make_directories
 	@echo "****** linking $(TSEVI_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(TSEVI_OLS) -o $@ $(TSEVI_LNK) 
+	$(COMPILADOR) $(FLAGS) $(TSEVI_OLS) -o $@ $(TSEVI_LNK) $(JUANFITS_OBJ)
 	@echo "## $(TSEVI_CLAU) linked successfully"
 
-$(TSEVI_OBJ) : $(TSEVI_SLS)
+$(TSEVI_OBJ) : $(TSEVI_SLS) make_directories
 	@echo "****** compiling $(TSEVI_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(TSEVI_SRC) $(TSEVI_INC) -o $@ 
 	@echo "## $(TSEVI_CLAU) compiled successfully"
@@ -1033,8 +675,8 @@
 #-----------------------------
 AN2ONYVA_CLAU = AN2_onyva
 AN2ONYVA_SRC = $(MAIN_PATH)/src/$(AN2ONYVA_CLAU).c
-AN2ONYVA_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(AN2ONYVA_CLAU).o
-AN2ONYVA_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(AN2ONYVA_CLAU).exe
+AN2ONYVA_OBJ = $(MAIN_PATH)/obj/$(AN2ONYVA_CLAU).o
+AN2ONYVA_EXE = $(MAIN_PATH)/bin/$(AN2ONYVA_CLAU).exe
 AN2ONYVA_SLS = $(AN2ONYVA_SRC) $(BT2_HDR)
 AN2ONYVA_OLS = $(AN2ONYVA_OBJ) 
 AN2ONYVA_LLS = $(BT2_LLS) 
@@ -1044,12 +686,12 @@
 $(AN2ONYVA_CLAU) : $(AN2ONYVA_EXE)
 	@echo " clau: $(AN2ONYVA_CLAU)"
 
-$(AN2ONYVA_EXE) : $(AN2ONYVA_OLS) $(AN2ONYVA_LLS)
+$(AN2ONYVA_EXE) : $(AN2ONYVA_OLS) $(AN2ONYVA_LLS) make_directories
 	@echo "****** linking $(AN2ONYVA_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(AN2ONYVA_OLS) -o $@ $(AN2ONYVA_LNK) 
+	$(COMPILADOR) $(FLAGS) $(AN2ONYVA_OLS) -o $@ $(AN2ONYVA_LNK) $(JUANFITS_OBJ)
 	@echo "## $(AN2ONYVA_CLAU) linked successfully"
 
-$(AN2ONYVA_OBJ) : $(AN2ONYVA_SLS)
+$(AN2ONYVA_OBJ) : $(AN2ONYVA_SLS) make_directories
 	@echo "****** compiling $(AN2ONYVA_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(AN2ONYVA_SRC) $(AN2ONYVA_INC) -o $@ 
 	@echo "## $(AN2ONYVA_CLAU) compiled successfully"
@@ -1060,8 +702,8 @@
 #-----------------------------
 WINID_CLAU = winid
 WINID_SRC = $(MAIN_PATH)/src/$(WINID_CLAU).c
-WINID_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(WINID_CLAU).o
-WINID_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(WINID_CLAU).exe
+WINID_OBJ = $(MAIN_PATH)/obj/$(WINID_CLAU).o
+WINID_EXE = $(MAIN_PATH)/bin/$(WINID_CLAU).exe
 WINID_SLS = $(WINID_SRC) $(BT2_HDR)
 WINID_OLS = $(WINID_OBJ) 
 WINID_LLS = $(BT2_LLS) 
@@ -1071,12 +713,12 @@
 $(WINID_CLAU) : $(WINID_EXE)
 	@echo " clau: $(WINID_CLAU)"
 
-$(WINID_EXE) : $(WINID_OLS) $(WINID_LLS)
+$(WINID_EXE) : $(WINID_OLS) $(WINID_LLS) make_directories
 	@echo "****** linking $(WINID_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(WINID_OLS) -o $@ $(WINID_LNK) 
+	$(COMPILADOR) $(FLAGS) $(WINID_OLS) -o $@ $(WINID_LNK) $(JUANFITS_OBJ) 
 	@echo "## $(WINID_CLAU) linked successfully"
 
-$(WINID_OBJ) : $(WINID_SLS)
+$(WINID_OBJ) : $(WINID_SLS) make_directories
 	@echo "****** compiling $(WINID_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(WINID_SRC) $(WINID_INC) -o $@ 
 	@echo "## $(WINID_CLAU) compiled successfully"
@@ -1087,8 +729,8 @@
 #-----------------------------
 HEADER_CLAU = header
 HEADER_SRC = $(MAIN_PATH)/src/$(HEADER_CLAU).c
-HEADER_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(HEADER_CLAU).o
-HEADER_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(HEADER_CLAU).exe
+HEADER_OBJ = $(MAIN_PATH)/obj/$(HEADER_CLAU).o
+HEADER_EXE = $(MAIN_PATH)/bin/$(HEADER_CLAU).exe
 HEADER_SLS = $(HEADER_SRC) $(BT2_HDR)
 HEADER_OLS = $(HEADER_OBJ) 
 HEADER_LLS = $(BT2_LLS) 
@@ -1098,12 +740,12 @@
 $(HEADER_CLAU) : $(HEADER_EXE)
 	@echo " clau: $(HEADER_CLAU)"
 
-$(HEADER_EXE) : $(HEADER_OLS) $(HEADER_LLS)
+$(HEADER_EXE) : $(HEADER_OLS) $(HEADER_LLS) make_directories
 	@echo "****** linking $(HEADER_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(HEADER_OLS) -o $@ $(HEADER_LNK) 
+	$(COMPILADOR) $(FLAGS) $(HEADER_OLS) -o $@ $(HEADER_LNK) $(JUANFITS_OBJ) 
 	@echo "## $(HEADER_CLAU) linked successfully"
 
-$(HEADER_OBJ) : $(HEADER_SLS)
+$(HEADER_OBJ) : $(HEADER_SLS) make_directories
 	@echo "****** compiling $(HEADER_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(HEADER_SRC) $(HEADER_INC) -o $@ 
 	@echo "## $(HEADER_CLAU) compiled successfully"
@@ -1114,8 +756,8 @@
 #-----------------------------
 LECTORIA_CLAU = lector_ia
 LECTORIA_SRC = $(MAIN_PATH)/src/$(LECTORIA_CLAU).c
-LECTORIA_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(LECTORIA_CLAU).o
-LECTORIA_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(LECTORIA_CLAU).exe
+LECTORIA_OBJ = $(MAIN_PATH)/obj/$(LECTORIA_CLAU).o
+LECTORIA_EXE = $(MAIN_PATH)/bin/$(LECTORIA_CLAU).exe
 LECTORIA_SLS = $(LECTORIA_SRC) $(BT2_HDR)
 LECTORIA_OLS = $(LECTORIA_OBJ) 
 LECTORIA_LLS = $(BT2_LLS) 
@@ -1125,12 +767,12 @@
 $(LECTORIA_CLAU) : $(LECTORIA_EXE)
 	@echo " clau: $(LECTORIA_CLAU)"
 
-$(LECTORIA_EXE) : $(LECTORIA_OLS) $(LECTORIA_LLS)
+$(LECTORIA_EXE) : $(LECTORIA_OLS) $(LECTORIA_LLS) make_directories
 	@echo "****** linking $(LECTORIA_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(LECTORIA_OLS) -o $@ $(LECTORIA_LNK) 
+	$(COMPILADOR) $(FLAGS) $(LECTORIA_OLS) -o $@ $(LECTORIA_LNK) $(JUANFITS_OBJ) -lcfitsio
 	@echo "## $(LECTORIA_CLAU) linked successfully"
 
-$(LECTORIA_OBJ) : $(LECTORIA_SLS)
+$(LECTORIA_OBJ) : $(LECTORIA_SLS) make_directories
 	@echo "****** compiling $(LECTORIA_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(LECTORIA_SRC) $(LECTORIA_INC) -o $@ 
 	@echo "## $(LECTORIA_CLAU) compiled successfully"
@@ -1141,8 +783,8 @@
 #-----------------------------
 BINARIAS_CLAU = binarias
 BINARIAS_SRC = $(MAIN_PATH)/src/$(BINARIAS_CLAU).c
-BINARIAS_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(BINARIAS_CLAU).o
-BINARIAS_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(BINARIAS_CLAU).exe
+BINARIAS_OBJ = $(MAIN_PATH)/obj/$(BINARIAS_CLAU).o
+BINARIAS_EXE = $(MAIN_PATH)/bin/$(BINARIAS_CLAU).exe
 BINARIAS_SLS = $(BINARIAS_SRC) $(BIN_HDR)
 BINARIAS_OLS = $(BINARIAS_OBJ) 
 BINARIAS_LLS = $(BIN_LLS) 
@@ -1152,12 +794,12 @@
 $(BINARIAS_CLAU) : $(BINARIAS_EXE)
 	@echo " clau: $(BINARIAS_CLAU)"
 
-$(BINARIAS_EXE) : $(BINARIAS_OLS) $(BINARIAS_LLS)
+$(BINARIAS_EXE) : $(BINARIAS_OLS) $(BINARIAS_LLS) make_directories
 	@echo "****** linking $(BINARIAS_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(BINARIAS_OLS) -o $@ $(BINARIAS_LNK) 
+	$(COMPILADOR) $(FLAGS) $(BINARIAS_OLS) -o $@ $(BINARIAS_LNK) $(JUANFITS_OBJ)
 	@echo "## $(BINARIAS_CLAU) linked successfully"
 
-$(BINARIAS_OBJ) : $(BINARIAS_SLS)
+$(BINARIAS_OBJ) : $(BINARIAS_SLS) make_directories
 	@echo "****** compiling $(BINARIAS_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(BINARIAS_SRC) $(BINARIAS_INC) -o $@ 
 	@echo "## $(BINARIAS_CLAU) compiled successfully"
@@ -1168,8 +810,8 @@
 #-----------------------------
 BINARIASFAKE_CLAU = binarias_fake
 BINARIASFAKE_SRC = $(MAIN_PATH)/src/$(BINARIASFAKE_CLAU).c
-BINARIASFAKE_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(BINARIASFAKE_CLAU).o
-BINARIASFAKE_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(BINARIASFAKE_CLAU).exe
+BINARIASFAKE_OBJ = $(MAIN_PATH)/obj/$(BINARIASFAKE_CLAU).o
+BINARIASFAKE_EXE = $(MAIN_PATH)/bin/$(BINARIASFAKE_CLAU).exe
 BINARIASFAKE_SLS = $(BINARIASFAKE_SRC) $(BIN_HDR)
 BINARIASFAKE_OLS = $(BINARIASFAKE_OBJ) 
 BINARIASFAKE_LLS = $(BIN_LLS) 
@@ -1179,12 +821,12 @@
 $(BINARIASFAKE_CLAU) : $(BINARIASFAKE_EXE)
 	@echo " clau: $(BINARIASFAKE_CLAU)"
 
-$(BINARIASFAKE_EXE) : $(BINARIASFAKE_OLS) $(BINARIASFAKE_LLS)
+$(BINARIASFAKE_EXE) : $(BINARIASFAKE_OLS) $(BINARIASFAKE_LLS) make_directories
 	@echo "****** linking $(BINARIASFAKE_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(BINARIASFAKE_OLS) -o $@ $(BINARIASFAKE_LNK) 
+	$(COMPILADOR) $(FLAGS) $(BINARIASFAKE_OLS) -o $@ $(BINARIASFAKE_LNK) $(JUANFITS_OBJ)
 	@echo "## $(BINARIASFAKE_CLAU) linked successfully"
 
-$(BINARIASFAKE_OBJ) : $(BINARIASFAKE_SLS)
+$(BINARIASFAKE_OBJ) : $(BINARIASFAKE_SLS) make_directories
 	@echo "****** compiling $(BINARIASFAKE_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(BINARIASFAKE_SRC) $(BINARIASFAKE_INC) -o $@ 
 	@echo "## $(BINARIASFAKE_CLAU) compiled successfully"
@@ -1196,8 +838,8 @@
 #-----------------------------
 TIMING_CLAU = timing
 TIMING_SRC = $(MAIN_PATH)/src/$(TIMING_CLAU).c
-TIMING_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(TIMING_CLAU).o
-TIMING_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(TIMING_CLAU).exe
+TIMING_OBJ = $(MAIN_PATH)/obj/$(TIMING_CLAU).o
+TIMING_EXE = $(MAIN_PATH)/bin/$(TIMING_CLAU).exe
 TIMING_SLS = $(TIMING_SRC) $(TMG_HDR)
 TIMING_OLS = $(TIMING_OBJ) 
 TIMING_LLS = $(TMG_LLS)
@@ -1207,12 +849,12 @@
 $(TIMING_CLAU) : $(TIMING_EXE)
 	@echo " clau: $(TIMING_CLAU)"
 
-$(TIMING_EXE) : $(TIMING_OLS) $(TIMING_LLS)
+$(TIMING_EXE) : $(TIMING_OLS) $(TIMING_LLS) make_directories
 	@echo "****** linking $(TIMING_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(TIMING_OLS) -o $@ $(TIMING_LNK) 
+	$(COMPILADOR) $(FLAGS) $(TIMING_OLS) -o $@ $(TIMING_LNK) $(JUANFITS_OBJ)
 	@echo "## $(TIMING_CLAU) linked successfully"
 
-$(TIMING_OBJ) : $(TIMING_SLS)
+$(TIMING_OBJ) : $(TIMING_SLS) make_directories
 	@echo "****** compiling $(TIMING_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(TIMING_SRC) $(TIMING_INC) -o $@ 
 	@echo "## $(TIMING_CLAU) compiled successfully"
@@ -1223,8 +865,8 @@
 #-----------------------------
 TIMINGAJTLIN_CLAU = timing_ajtlin
 TIMINGAJTLIN_SRC = $(MAIN_PATH)/src/$(TIMINGAJTLIN_CLAU).c
-TIMINGAJTLIN_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(TIMINGAJTLIN_CLAU).o
-TIMINGAJTLIN_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(TIMINGAJTLIN_CLAU).exe
+TIMINGAJTLIN_OBJ = $(MAIN_PATH)/obj/$(TIMINGAJTLIN_CLAU).o
+TIMINGAJTLIN_EXE = $(MAIN_PATH)/bin/$(TIMINGAJTLIN_CLAU).exe
 TIMINGAJTLIN_SLS = $(TIMINGAJTLIN_SRC) $(TMG_HDR)
 TIMINGAJTLIN_OLS = $(TIMINGAJTLIN_OBJ) 
 TIMINGAJTLIN_LLS = $(TMG_LLS)
@@ -1234,12 +876,12 @@
 $(TIMINGAJTLIN_CLAU) : $(TIMINGAJTLIN_EXE)
 	@echo " clau: $(TIMINGAJTLIN_CLAU)"
 
-$(TIMINGAJTLIN_EXE) : $(TIMINGAJTLIN_OLS) $(TIMINGAJTLIN_LLS)
+$(TIMINGAJTLIN_EXE) : $(TIMINGAJTLIN_OLS) $(TIMINGAJTLIN_LLS) make_directories
 	@echo "****** linking $(TIMINGAJTLIN_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(TIMINGAJTLIN_OLS) -o $@ $(TIMINGAJTLIN_LNK) 
+	$(COMPILADOR) $(FLAGS) $(TIMINGAJTLIN_OLS) -o $@ $(TIMINGAJTLIN_LNK) $(JUANFITS_OBJ)
 	@echo "## $(TIMINGAJTLIN_CLAU) linked successfully"
 
-$(TIMINGAJTLIN_OBJ) : $(TIMINGAJTLIN_SLS)
+$(TIMINGAJTLIN_OBJ) : $(TIMINGAJTLIN_SLS) make_directories
 	@echo "****** compiling $(TIMINGAJTLIN_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(TIMINGAJTLIN_SRC) $(TIMINGAJTLIN_INC) -o $@ 
 	@echo "## $(TIMINGAJTLIN_CLAU) compiled successfully"
@@ -1250,8 +892,8 @@
 #-----------------------------
 TIMINGIRWIN_CLAU = timing_irwin
 TIMINGIRWIN_SRC = $(MAIN_PATH)/src/$(TIMINGIRWIN_CLAU).c
-TIMINGIRWIN_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(TIMINGIRWIN_CLAU).o
-TIMINGIRWIN_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(TIMINGIRWIN_CLAU).exe
+TIMINGIRWIN_OBJ = $(MAIN_PATH)/obj/$(TIMINGIRWIN_CLAU).o
+TIMINGIRWIN_EXE = $(MAIN_PATH)/bin/$(TIMINGIRWIN_CLAU).exe
 TIMINGIRWIN_SLS = $(TIMINGIRWIN_SRC) $(TMG_HDR)
 TIMINGIRWIN_OLS = $(TIMINGIRWIN_OBJ) 
 TIMINGIRWIN_LLS = $(TMG_LLS)
@@ -1261,12 +903,12 @@
 $(TIMINGIRWIN_CLAU) : $(TIMINGIRWIN_EXE)
 	@echo " clau: $(TIMINGIRWIN_CLAU)"
 
-$(TIMINGIRWIN_EXE) : $(TIMINGIRWIN_OLS) $(TIMINGIRWIN_LLS)
+$(TIMINGIRWIN_EXE) : $(TIMINGIRWIN_OLS) $(TIMINGIRWIN_LLS) make_directories
 	@echo "****** linking $(TIMINGIRWIN_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(TIMINGIRWIN_OLS) -o $@ $(TIMINGIRWIN_LNK) 
+	$(COMPILADOR) $(FLAGS) $(TIMINGIRWIN_OLS) -o $@ $(TIMINGIRWIN_LNK) $(JUANFITS_OBJ) -lcfitsio
 	@echo "## $(TIMINGIRWIN_CLAU) linked successfully"
 
-$(TIMINGIRWIN_OBJ) : $(TIMINGIRWIN_SLS)
+$(TIMINGIRWIN_OBJ) : $(TIMINGIRWIN_SLS) make_directories
 	@echo "****** compiling $(TIMINGIRWIN_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(TIMINGIRWIN_SRC) $(TIMINGIRWIN_INC) -o $@ 
 	@echo "## $(TIMINGIRWIN_CLAU) compiled successfully"
@@ -1277,8 +919,8 @@
 #-----------------------------
 AN2TIMING_CLAU = AN2_timing
 AN2TIMING_SRC = $(MAIN_PATH)/src/$(AN2TIMING_CLAU).c
-AN2TIMING_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(AN2TIMING_CLAU).o
-AN2TIMING_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(AN2TIMING_CLAU).exe
+AN2TIMING_OBJ = $(MAIN_PATH)/obj/$(AN2TIMING_CLAU).o
+AN2TIMING_EXE = $(MAIN_PATH)/bin/$(AN2TIMING_CLAU).exe
 AN2TIMING_SLS = $(AN2TIMING_SRC) $(TMG_HDR)
 AN2TIMING_OLS = $(AN2TIMING_OBJ) 
 AN2TIMING_LLS = $(TMG_LLS)
@@ -1288,12 +930,12 @@
 $(AN2TIMING_CLAU) : $(AN2TIMING_EXE)
 	@echo " clau: $(AN2TIMING_CLAU)"
 
-$(AN2TIMING_EXE) : $(AN2TIMING_OLS) $(AN2TIMING_LLS)
+$(AN2TIMING_EXE) : $(AN2TIMING_OLS) $(AN2TIMING_LLS) make_directories
 	@echo "****** linking $(AN2TIMING_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(AN2TIMING_OLS) -o $@ $(AN2TIMING_LNK) 
+	$(COMPILADOR) $(FLAGS) $(AN2TIMING_OLS) -o $@ $(AN2TIMING_LNK) $(JUANFITS_OBJ)
 	@echo "## $(AN2TIMING_CLAU) linked successfully"
 
-$(AN2TIMING_OBJ) : $(AN2TIMING_SLS)
+$(AN2TIMING_OBJ) : $(AN2TIMING_SLS) make_directories
 	@echo "****** compiling $(AN2TIMING_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(AN2TIMING_SRC) $(AN2TIMING_INC) -o $@ 
 	@echo "## $(AN2TIMING_CLAU) compiled successfully"
@@ -1304,8 +946,8 @@
 #-----------------------------
 FITMODPL_CLAU = fitmodpl
 FITMODPL_SRC = $(MAIN_PATH)/src/$(FITMODPL_CLAU).c
-FITMODPL_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(FITMODPL_CLAU).o
-FITMODPL_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(FITMODPL_CLAU).exe
+FITMODPL_OBJ = $(MAIN_PATH)/obj/$(FITMODPL_CLAU).o
+FITMODPL_EXE = $(MAIN_PATH)/bin/$(FITMODPL_CLAU).exe
 FITMODPL_SLS = $(FITMODPL_SRC) $(MPL_HDR)
 FITMODPL_OLS = $(FITMODPL_OBJ) 
 FITMODPL_LLS = $(MPL_LLS) 
@@ -1315,12 +957,12 @@
 $(FITMODPL_CLAU) : $(FITMODPL_EXE)
 	@echo " clau: $(FITMODPL_CLAU)"
 
-$(FITMODPL_EXE) : $(FITMODPL_OLS) $(FITMODPL_LLS)
+$(FITMODPL_EXE) : $(FITMODPL_OLS) $(FITMODPL_LLS) make_directories
 	@echo "****** linking $(FITMODPL_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(FITMODPL_OLS) -o $@ $(FITMODPL_LNK) 
+	$(COMPILADOR) $(FLAGS) $(FITMODPL_OLS) -o $@ $(FITMODPL_LNK) $(JUANFITS_OBJ)
 	@echo "## $(FITMODPL_CLAU) linked successfully"
 
-$(FITMODPL_OBJ) : $(FITMODPL_SLS)
+$(FITMODPL_OBJ) : $(FITMODPL_SLS) make_directories
 	@echo "****** compiling $(FITMODPL_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(FITMODPL_SRC) $(FITMODPL_INC) -o $@ 
 	@echo "## $(FITMODPL_CLAU) compiled successfully"
@@ -1331,8 +973,8 @@
 #-----------------------------
 DATEJD_CLAU = datejd
 DATEJD_SRC = $(MAIN_PATH)/src/$(DATEJD_CLAU).c
-DATEJD_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(DATEJD_CLAU).o
-DATEJD_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(DATEJD_CLAU).exe
+DATEJD_OBJ = $(MAIN_PATH)/obj/$(DATEJD_CLAU).o
+DATEJD_EXE = $(MAIN_PATH)/bin/$(DATEJD_CLAU).exe
 DATEJD_SLS = $(DATEJD_SRC) $(JUANFITS_HDR)
 DATEJD_OLS = $(DATEJD_OBJ) 
 DATEJD_LLS = $(JUANFITS_LLS) 
@@ -1342,12 +984,12 @@
 $(DATEJD_CLAU) : $(DATEJD_EXE)
 	@echo " clau: $(DATEJD_CLAU)"
 
-$(DATEJD_EXE) : $(DATEJD_OLS) $(DATEJD_LLS)
+$(DATEJD_EXE) : $(DATEJD_OLS) $(DATEJD_LLS) make_directories
 	@echo "****** linking $(DATEJD_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(DATEJD_OLS) -o $@ $(DATEJD_LNK) 
+	$(COMPILADOR) $(FLAGS) $(DATEJD_OLS) -o $@ $(DATEJD_LNK) $(JUANFITS_OBJ)
 	@echo "## $(DATEJD_CLAU) linked successfully"
 
-$(DATEJD_OBJ) : $(DATEJD_SLS)
+$(DATEJD_OBJ) : $(DATEJD_SLS) make_directories
 	@echo "****** compiling $(DATEJD_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(DATEJD_SRC) $(DATEJD_INC) -o $@ 
 	@echo "## $(DATEJD_CLAU) compiled successfully"
@@ -1359,8 +1001,8 @@
 #-----------------------------
 TENDENCIA_CLAU = tendencia
 TENDENCIA_SRC = $(MAIN_PATH)/src/$(TENDENCIA_CLAU).c
-TENDENCIA_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(TENDENCIA_CLAU).o
-TENDENCIA_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(TENDENCIA_CLAU).exe
+TENDENCIA_OBJ = $(MAIN_PATH)/obj/$(TENDENCIA_CLAU).o
+TENDENCIA_EXE = $(MAIN_PATH)/bin/$(TENDENCIA_CLAU).exe
 TENDENCIA_SLS = $(TENDENCIA_SRC) $(BT2_HDR)
 TENDENCIA_OLS = $(TENDENCIA_OBJ) 
 TENDENCIA_LLS = $(BT2_LLS) 
@@ -1370,12 +1012,12 @@
 $(TENDENCIA_CLAU) : $(TENDENCIA_EXE)
 	@echo " clau: $(TENDENCIA_CLAU)"
 
-$(TENDENCIA_EXE) : $(TENDENCIA_OLS) $(TENDENCIA_LLS)
+$(TENDENCIA_EXE) : $(TENDENCIA_OLS) $(TENDENCIA_LLS) make_directories
 	@echo "****** linking $(TENDENCIA_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(TENDENCIA_OLS) -o $@ $(TENDENCIA_LNK) 
+	$(COMPILADOR) $(FLAGS) $(TENDENCIA_OLS) -o $@ $(TENDENCIA_LNK) $(JUANFITS_OBJ)
 	@echo "## $(TENDENCIA_CLAU) linked successfully"
 
-$(TENDENCIA_OBJ) : $(TENDENCIA_SLS)
+$(TENDENCIA_OBJ) : $(TENDENCIA_SLS) make_directories
 	@echo "****** compiling $(TENDENCIA_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(TENDENCIA_SRC) $(TENDENCIA_INC) -o $@ 
 	@echo "## $(TENDENCIA_CLAU) compiled successfully"
@@ -1386,8 +1028,8 @@
 #-----------------------------
 FOLD_CLAU = fold
 FOLD_SRC = $(MAIN_PATH)/src/$(FOLD_CLAU).c
-FOLD_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(FOLD_CLAU).o
-FOLD_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(FOLD_CLAU).exe
+FOLD_OBJ = $(MAIN_PATH)/obj/$(FOLD_CLAU).o
+FOLD_EXE = $(MAIN_PATH)/bin/$(FOLD_CLAU).exe
 FOLD_SLS = $(FOLD_SRC) $(BT2_HDR)
 FOLD_OLS = $(FOLD_OBJ) 
 FOLD_LLS = $(BT2_LLS) 
@@ -1397,12 +1039,12 @@
 $(FOLD_CLAU) : $(FOLD_EXE)
 	@echo " clau: $(FOLD_CLAU)"
 
-$(FOLD_EXE) : $(FOLD_OLS) $(FOLD_LLS)
+$(FOLD_EXE) : $(FOLD_OLS) $(FOLD_LLS) make_directories
 	@echo "****** linking $(FOLD_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(FOLD_OLS) -o $@ $(FOLD_LNK) 
+	$(COMPILADOR) $(FLAGS) $(FOLD_OLS) -o $@ $(FOLD_LNK) $(JUANFITS_OBJ)
 	@echo "## $(FOLD_CLAU) linked successfully"
 
-$(FOLD_OBJ) : $(FOLD_SLS)
+$(FOLD_OBJ) : $(FOLD_SLS) make_directories
 	@echo "****** compiling $(FOLD_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(FOLD_SRC) $(FOLD_INC) -o $@ 
 	@echo "## $(FOLD_CLAU) compiled successfully"
@@ -1413,8 +1055,8 @@
 #-----------------------------
 TRADUCE_CLAU = traduce
 TRADUCE_SRC = $(MAIN_PATH)/src/$(TRADUCE_CLAU).c
-TRADUCE_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(TRADUCE_CLAU).o
-TRADUCE_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(TRADUCE_CLAU).exe
+TRADUCE_OBJ = $(MAIN_PATH)/obj/$(TRADUCE_CLAU).o
+TRADUCE_EXE = $(MAIN_PATH)/bin/$(TRADUCE_CLAU).exe
 TRADUCE_SLS = $(TRADUCE_SRC) 
 TRADUCE_OLS = $(TRADUCE_OBJ) 
 TRADUCE_LLS = 
@@ -1424,12 +1066,12 @@
 $(TRADUCE_CLAU) : $(TRADUCE_EXE)
 	@echo " clau: $(TRADUCE_CLAU)"
 
-$(TRADUCE_EXE) : $(TRADUCE_OLS) $(TRADUCE_LLS)
+$(TRADUCE_EXE) : $(TRADUCE_OLS) $(TRADUCE_LLS) make_directories
 	@echo "****** linking $(TRADUCE_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(TRADUCE_OLS) -o $@ $(TRADUCE_LNK) 
+	$(COMPILADOR) $(FLAGS) $(TRADUCE_OLS) -o $@ $(TRADUCE_LNK) $(JUANFITS_OBJ) -lcfitsio
 	@echo "## $(TRADUCE_CLAU) linked successfully"
 
-$(TRADUCE_OBJ) : $(TRADUCE_SLS)
+$(TRADUCE_OBJ) : $(TRADUCE_SLS) make_directories
 	@echo "****** compiling $(TRADUCE_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(TRADUCE_SRC) $(TRADUCE_INC) -o $@ 
 	@echo "## $(TRADUCE_CLAU) compiled successfully"
@@ -1440,8 +1082,8 @@
 #-----------------------------
 HORRIBLESOSPECHA_CLAU = horriblesospecha
 HORRIBLESOSPECHA_SRC = $(MAIN_PATH)/src/$(HORRIBLESOSPECHA_CLAU).c
-HORRIBLESOSPECHA_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(HORRIBLESOSPECHA_CLAU).o
-HORRIBLESOSPECHA_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(HORRIBLESOSPECHA_CLAU).exe
+HORRIBLESOSPECHA_OBJ = $(MAIN_PATH)/obj/$(HORRIBLESOSPECHA_CLAU).o
+HORRIBLESOSPECHA_EXE = $(MAIN_PATH)/bin/$(HORRIBLESOSPECHA_CLAU).exe
 HORRIBLESOSPECHA_SLS = $(HORRIBLESOSPECHA_SRC) $(BT2_HDR)
 HORRIBLESOSPECHA_OLS = $(HORRIBLESOSPECHA_OBJ) 
 HORRIBLESOSPECHA_LLS = $(BT2_LLS) 
@@ -1451,12 +1093,12 @@
 $(HORRIBLESOSPECHA_CLAU) : $(HORRIBLESOSPECHA_EXE)
 	@echo " clau: $(HORRIBLESOSPECHA_CLAU)"
 
-$(HORRIBLESOSPECHA_EXE) : $(HORRIBLESOSPECHA_OLS) $(HORRIBLESOSPECHA_LLS)
+$(HORRIBLESOSPECHA_EXE) : $(HORRIBLESOSPECHA_OLS) $(HORRIBLESOSPECHA_LLS) make_directories
 	@echo "****** linking $(HORRIBLESOSPECHA_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(HORRIBLESOSPECHA_OLS) -o $@ $(HORRIBLESOSPECHA_LNK) 
+	$(COMPILADOR) $(FLAGS) $(HORRIBLESOSPECHA_OLS) -o $@ $(HORRIBLESOSPECHA_LNK) $(JUANFITS_OBJ)
 	@echo "## $(HORRIBLESOSPECHA_CLAU) linked successfully"
 
-$(HORRIBLESOSPECHA_OBJ) : $(HORRIBLESOSPECHA_SLS)
+$(HORRIBLESOSPECHA_OBJ) : $(HORRIBLESOSPECHA_SLS) make_directories
 	@echo "****** compiling $(HORRIBLESOSPECHA_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(HORRIBLESOSPECHA_SRC) $(HORRIBLESOSPECHA_INC) -o $@ 
 	@echo "## $(HORRIBLESOSPECHA_CLAU) compiled successfully"
@@ -1467,8 +1109,8 @@
 #-----------------------------
 REDNOISE_CLAU = rednoise
 REDNOISE_SRC = $(MAIN_PATH)/src/$(REDNOISE_CLAU).c
-REDNOISE_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(REDNOISE_CLAU).o
-REDNOISE_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(REDNOISE_CLAU).exe
+REDNOISE_OBJ = $(MAIN_PATH)/obj/$(REDNOISE_CLAU).o
+REDNOISE_EXE = $(MAIN_PATH)/bin/$(REDNOISE_CLAU).exe
 REDNOISE_SLS = $(REDNOISE_SRC) $(BT2_HDR)
 REDNOISE_OLS = $(REDNOISE_OBJ) 
 REDNOISE_LLS = $(BT2_LLS) 
@@ -1478,12 +1120,12 @@
 $(REDNOISE_CLAU) : $(REDNOISE_EXE)
 	@echo " clau: $(REDNOISE_CLAU)"
 
-$(REDNOISE_EXE) : $(REDNOISE_OLS) $(REDNOISE_LLS)
+$(REDNOISE_EXE) : $(REDNOISE_OLS) $(REDNOISE_LLS) make_directories
 	@echo "****** linking $(REDNOISE_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(REDNOISE_OLS) -o $@ $(REDNOISE_LNK) 
+	$(COMPILADOR) $(FLAGS) $(REDNOISE_OLS) -o $@ $(REDNOISE_LNK) $(JUANFITS_OBJ)
 	@echo "## $(REDNOISE_CLAU) linked successfully"
 
-$(REDNOISE_OBJ) : $(REDNOISE_SLS)
+$(REDNOISE_OBJ) : $(REDNOISE_SLS) make_directories
 	@echo "****** compiling $(REDNOISE_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(REDNOISE_SRC) $(REDNOISE_INC) -o $@ 
 	@echo "## $(REDNOISE_CLAU) compiled successfully"
@@ -1494,8 +1136,8 @@
 #-----------------------------
 WDSCDB_CLAU = wdscdb
 WDSCDB_SRC = $(MAIN_PATH)/src/$(WDSCDB_CLAU).c
-WDSCDB_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(WDSCDB_CLAU).o
-WDSCDB_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(WDSCDB_CLAU).exe
+WDSCDB_OBJ = $(MAIN_PATH)/obj/$(WDSCDB_CLAU).o
+WDSCDB_EXE = $(MAIN_PATH)/bin/$(WDSCDB_CLAU).exe
 WDSCDB_SLS = $(WDSCDB_SRC) $(JUAN_HDR) $(JUANFITS_HDR)
 WDSCDB_OLS = $(WDSCDB_OBJ) 
 WDSCDB_LLS = $(JUAN_LLS) $(JUANFITS_LLS) 
@@ -1505,12 +1147,12 @@
 $(WDSCDB_CLAU) : $(WDSCDB_EXE)
 	@echo " clau: $(WDSCDB_CLAU)"
 
-$(WDSCDB_EXE) : $(WDSCDB_OLS) $(WDSCDB_LLS)
+$(WDSCDB_EXE) : $(WDSCDB_OLS) $(WDSCDB_LLS) make_directories
 	@echo "****** linking $(WDSCDB_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(WDSCDB_OLS) -o $@ $(WDSCDB_LNK) 
+	$(COMPILADOR) $(FLAGS) $(WDSCDB_OLS) -o $@ $(WDSCDB_LNK) $(JUANFITS_OBJ)
 	@echo "## $(WDSCDB_CLAU) linked successfully"
 
-$(WDSCDB_OBJ) : $(WDSCDB_SLS)
+$(WDSCDB_OBJ) : $(WDSCDB_SLS) make_directories
 	@echo "****** compiling $(WDSCDB_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(WDSCDB_SRC) $(WDSCDB_INC) -o $@ 
 	@echo "## $(WDSCDB_CLAU) compiled successfully"
@@ -1521,8 +1163,8 @@
 #-----------------------------
 READCLNSET_CLAU = readclnset
 READCLNSET_SRC = $(MAIN_PATH)/src/$(READCLNSET_CLAU).c
-READCLNSET_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(READCLNSET_CLAU).o
-READCLNSET_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(READCLNSET_CLAU).exe
+READCLNSET_OBJ = $(MAIN_PATH)/obj/$(READCLNSET_CLAU).o
+READCLNSET_EXE = $(MAIN_PATH)/bin/$(READCLNSET_CLAU).exe
 READCLNSET_SLS = $(READCLNSET_SRC) $(JUAN_HDR) $(JUANFITS_HDR)
 READCLNSET_OLS = $(READCLNSET_OBJ) 
 READCLNSET_LLS = $(JUAN_LLS) $(JUANFITS_LLS) 
@@ -1532,12 +1174,12 @@
 $(READCLNSET_CLAU) : $(READCLNSET_EXE)
 	@echo " clau: $(READCLNSET_CLAU)"
 
-$(READCLNSET_EXE) : $(READCLNSET_OLS) $(READCLNSET_LLS)
+$(READCLNSET_EXE) : $(READCLNSET_OLS) $(READCLNSET_LLS) make_directories
 	@echo "****** linking $(READCLNSET_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(READCLNSET_OLS) -o $@ $(READCLNSET_LNK) 
+	$(COMPILADOR) $(FLAGS) $(READCLNSET_OLS) -o $@ $(READCLNSET_LNK) $(JUANFITS_OBJ)
 	@echo "## $(READCLNSET_CLAU) linked successfully"
 
-$(READCLNSET_OBJ) : $(READCLNSET_SLS)
+$(READCLNSET_OBJ) : $(READCLNSET_SLS) make_directories
 	@echo "****** compiling $(READCLNSET_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(READCLNSET_SRC) $(READCLNSET_INC) -o $@ 
 	@echo "## $(READCLNSET_CLAU) compiled successfully"
@@ -1548,8 +1190,8 @@
 #-----------------------------
 HSA_CLAU = hsa
 HSA_SRC = $(MAIN_PATH)/src/$(HSA_CLAU).c
-HSA_OBJ = $(MAIN_PATH)/obj/$(ORDENA)/$(HSA_CLAU).o
-HSA_EXE = $(MAIN_PATH)/bin/$(ORDENA)/$(HSA_CLAU).exe
+HSA_OBJ = $(MAIN_PATH)/obj/$(HSA_CLAU).o
+HSA_EXE = $(MAIN_PATH)/bin/$(HSA_CLAU).exe
 HSA_SLS = $(HSA_SRC)
 HSA_OLS = $(HSA_OBJ) 
 HSA_LLS = 
@@ -1559,12 +1201,12 @@
 $(HSA_CLAU) : $(HSA_EXE)
 	@echo " clau: $(HSA_CLAU)"
 
-$(HSA_EXE) : $(HSA_OLS) $(HSA_LLS)
+$(HSA_EXE) : $(HSA_OLS) $(HSA_LLS) make_directories
 	@echo "****** linking $(HSA_CLAU) *****"
-	$(COMPILADOR) $(FLAGS) $(HSA_OLS) -o $@ $(HSA_LNK) 
+	$(COMPILADOR) $(FLAGS) $(HSA_OLS) -o $@ $(HSA_LNK) $(JUANFITS_OBJ) -lcfitsio
 	@echo "## $(HSA_CLAU) linked successfully"
 
-$(HSA_OBJ) : $(HSA_SLS)
+$(HSA_OBJ) : $(HSA_SLS) make_directories
 	@echo "****** compiling $(HSA_CLAU) *****"
 	$(COMPILADOR) $(FLAGS) -c $(HSA_SRC) $(HSA_INC) -o $@ 
 	@echo "## $(HSA_CLAU) compiled successfully"
@@ -1573,13 +1215,13 @@
 
 # libreria juan
 #-----------------------------
-$(JUAN_LIB) : $(JUAN_OBJ)
+$(JUAN_LIB) : $(JUAN_OBJ) make_directories
 	@echo "***** creating library $(JUAN_CLAU) *****"
 	ar -rc $@ $(JUAN_OBJ)
 	ranlib $@
 	@echo "## library created successfully"
 
-$(JUAN_OBJ) : $(JUAN_SLS) 
+$(JUAN_OBJ) : $(JUAN_SLS)  make_directories
 	@echo "***** compiling $(JUAN_CLAU) *****"
 	$(COMPILADOR) -o $@ -c $(JUAN_SRC) $(JUAN_INC)
 	@echo "## library compiled successfully"
@@ -1591,13 +1233,13 @@
 
 # libreria juan_fits
 #-----------------------------
-$(JUANFITS_LIB) : $(JUANFITS_OBJ)
+$(JUANFITS_LIB) : $(JUANFITS_OBJ) make_directories
 	@echo "***** creating library $(JUANFITS_CLAU) *****"
 	ar -rc $@ $(JUANFITS_OBJ)
 	ranlib $@
 	@echo "## library created successfully"
 
-$(JUANFITS_OBJ) : $(JUANFITS_SLS)
+$(JUANFITS_OBJ) : $(JUANFITS_SLS) make_directories
 	@echo "***** compiling $(JUANFITS_CLAU) *****"
 	$(COMPILADOR) -o $@ -c $(JUANFITS_SRC) $(JUANFITS_INC)
 	@echo "## library compiled successfully"
@@ -1609,13 +1251,13 @@
 
 # libreria libbt2
 #-----------------------------
-$(BT2_LIB) : $(BT2_OBJ)
+$(BT2_LIB) : $(BT2_OBJ) make_directories
 	@echo "***** creating library $(BT2_CLAU) *****"
 	ar -rc $@ $(BT2_OBJ)
 	ranlib $@
 	@echo "## library created successfully"
 
-$(BT2_OBJ) : $(BT2_SLS) 
+$(BT2_OBJ) : $(BT2_SLS)  make_directories
 	@echo "***** compiling $(BT2_CLAU) *****"
 	$(COMPILADOR) -o $@ -c $(BT2_SRC) $(BT2_INC)
 	@echo "## library compiled successfully"
@@ -1627,13 +1269,13 @@
 
 # libreria libbt2_clnset
 #-----------------------------
-$(BT2CLNSET_LIB) : $(BT2CLNSET_OBJ)
+$(BT2CLNSET_LIB) : $(BT2CLNSET_OBJ) make_directories
 	@echo "***** creating library $(BT2CLNSET_CLAU) *****"
 	ar -rc $@ $(BT2CLNSET_OBJ)
 	ranlib $@
 	@echo "## library created successfully"
 
-$(BT2CLNSET_OBJ) : $(BT2CLNSET_SLS) 
+$(BT2CLNSET_OBJ) : $(BT2CLNSET_SLS)  make_directories
 	@echo "***** compiling $(BT2CLNSET_CLAU) *****"
 	$(COMPILADOR) -o $@ -c $(BT2CLNSET_SRC) $(BT2CLNSET_INC)
 	@echo "## library compiled successfully"
@@ -1645,13 +1287,13 @@
 
 # libreria libbin
 #-----------------------------
-$(BIN_LIB) : $(BIN_OBJ)
+$(BIN_LIB) : $(BIN_OBJ) make_directories
 	@echo "***** creating library $(BIN_CLAU) *****"
 	ar -rc $@ $(BIN_OBJ)
 	ranlib $@
 	@echo "## library created successfully"
 
-$(BIN_OBJ) : $(BIN_SLS) 
+$(BIN_OBJ) : $(BIN_SLS)  make_directories
 	@echo "***** compiling $(BIN_CLAU) *****"
 	$(COMPILADOR) -o $@ -c $(BIN_SRC) $(BIN_INC)
 	@echo "## library compiled successfully"
@@ -1663,13 +1305,13 @@
 
 # libreria libtmg
 #-----------------------------
-$(TMG_LIB) : $(TMG_OBJ)
+$(TMG_LIB) : $(TMG_OBJ) make_directories
 	@echo "***** creating library $(TMG_CLAU) *****"
 	ar -rc $@ $(TMG_OBJ)
 	ranlib $@
 	@echo "## library created successfully"
 
-$(TMG_OBJ) : $(TMG_SLS) 
+$(TMG_OBJ) : $(TMG_SLS)  make_directories
 	@echo "***** compiling $(TMG_CLAU) *****"
 	$(COMPILADOR) -o $@ -c $(TMG_SRC) $(TMG_INC)
 	@echo "## library compiled successfully"
@@ -1681,13 +1323,13 @@
 
 # libreria libmpl
 #-----------------------------
-$(MPL_LIB) : $(MPL_OBJ)
+$(MPL_LIB) : $(MPL_OBJ) make_directories
 	@echo "***** creating library $(MPL_CLAU) *****"
 	ar -rc $@ $(MPL_OBJ)
 	ranlib $@
 	@echo "## library created successfully"
 
-$(MPL_OBJ) : $(MPL_SLS) 
+$(MPL_OBJ) : $(MPL_SLS)  make_directories
 	@echo "***** compiling $(MPL_CLAU) *****"
 	$(COMPILADOR) -o $@ -c $(MPL_SRC) $(MPL_INC)
 	@echo "## library compiled successfully"
@@ -1699,15 +1341,15 @@
 
 # library libkepler_fits
 #-----------------------------
-$(KFTS_LIB) : $(KFTS_OBJ)
+$(KFTS_LIB) : $(KFTS_OBJ) make_directories
 	@echo "***** creating library $(KFTS_CLAU) *****"
 	ar -rc $@ $(KFTS_OBJ)
 	ranlib $@
 	@echo "## library created successfully"
 
-$(KFTS_OBJ) : $(KFTS_SRC) $(KFTS_HDR_LIST) 
+$(KFTS_OBJ) : $(KFTS_SRC) $(KFTS_HDR_LIST)  make_directories
 	@echo "***** compiling $(KFTS_CLAU) *****"
-	$(COMPILADOR) -o $@ -c $(KFTS_SRC) $(KFTS_INC)
+	$(COMPILADOR) -o $@ -c $(KFTS_SRC) $(KFTS_INC) -I$(JUANFITS_PATH)
 	@echo "## library compiled successfully"
 
 $(KFTS_SRC) :
@@ -1717,15 +1359,15 @@
 
 # library libk2vanderburg_fits
 #-----------------------------
-$(K2VFTS_LIB) : $(K2VFTS_OBJ)
+$(K2VFTS_LIB) : $(K2VFTS_OBJ) make_directories
 	@echo "***** creating library $(K2VFTS_CLAU) *****"
 	ar -rc $@ $(K2VFTS_OBJ)
 	ranlib $@
 	@echo "## library created successfully"
 
-$(K2VFTS_OBJ) : $(K2VFTS_SRC) $(K2VFTS_HDR_LIST) 
+$(K2VFTS_OBJ) : $(K2VFTS_SRC) $(K2VFTS_HDR_LIST)  make_directories
 	@echo "***** compiling $(K2VFTS_CLAU) *****"
-	$(COMPILADOR) -o $@ -c $(K2VFTS_SRC) $(K2VFTS_INC)
+	$(COMPILADOR) -o $@ -c $(K2VFTS_SRC) $(K2VFTS_INC) -I$(JUANFITS_PATH)
 	@echo "## library compiled successfully"
 
 $(K2VFTS_SRC) :
@@ -1735,15 +1377,15 @@
 
 # library libeverest_fits
 #-----------------------------
-$(K2EFTS_LIB) : $(K2EFTS_OBJ)
+$(K2EFTS_LIB) : $(K2EFTS_OBJ) make_directories
 	@echo "***** creating library $(K2EFTS_CLAU) *****"
 	ar -rc $@ $(K2EFTS_OBJ)
 	ranlib $@
 	@echo "## library created successfully"
 
-$(K2EFTS_OBJ) : $(K2EFTS_SRC) $(K2EFTS_HDR_LIST) 
+$(K2EFTS_OBJ) : $(K2EFTS_SRC) $(K2EFTS_HDR_LIST)  make_directories
 	@echo "***** compiling $(K2EFTS_CLAU) *****"
-	$(COMPILADOR) -o $@ -c $(K2EFTS_SRC) $(K2EFTS_INC)
+	$(COMPILADOR) -o $@ -c $(K2EFTS_SRC) $(K2EFTS_INC) -I$(JUANFITS_PATH)
 	@echo "## library compiled successfully"
 
 $(K2EFTS_SRC) :
@@ -1753,15 +1395,15 @@
 
 # library libk2sc_fits
 #-----------------------------
-$(K2SCFTS_LIB) : $(K2SCFTS_OBJ)
+$(K2SCFTS_LIB) : $(K2SCFTS_OBJ) make_directories
 	@echo "***** creating library $(K2SCFTS_CLAU) *****"
 	ar -rc $@ $(K2SCFTS_OBJ)
 	ranlib $@
 	@echo "## library created successfully"
 
-$(K2SCFTS_OBJ) : $(K2SCFTS_SRC) $(K2SCFTS_HDR_LIST) 
+$(K2SCFTS_OBJ) : $(K2SCFTS_SRC) $(K2SCFTS_HDR_LIST)  make_directories
 	@echo "***** compiling $(K2SCFTS_CLAU) *****"
-	$(COMPILADOR) -o $@ -c $(K2SCFTS_SRC) $(K2SCFTS_INC)
+	$(COMPILADOR) -o $@ -c $(K2SCFTS_SRC) $(K2SCFTS_INC) -I$(JUANFITS_PATH)
 	@echo "## library compiled successfully"
 
 $(K2SCFTS_SRC) :
@@ -1771,15 +1413,15 @@
 
 # library libpolar_fits
 #-----------------------------
-$(K2PFTS_LIB) : $(K2PFTS_OBJ)
+$(K2PFTS_LIB) : $(K2PFTS_OBJ) make_directories
 	@echo "***** creating library $(K2PFTS_CLAU) *****"
 	ar -rc $@ $(K2PFTS_OBJ)
 	ranlib $@
 	@echo "## library created successfully"
 
-$(K2PFTS_OBJ) : $(K2PFTS_SRC) $(K2PFTS_HDR_LIST) 
+$(K2PFTS_OBJ) : $(K2PFTS_SRC) $(K2PFTS_HDR_LIST)  make_directories
 	@echo "***** compiling $(K2PFTS_CLAU) *****"
-	$(COMPILADOR) -o $@ -c $(K2PFTS_SRC) $(K2PFTS_INC)
+	$(COMPILADOR) -o $@ -c $(K2PFTS_SRC) $(K2PFTS_INC) -I$(JUANFITS_PATH)
 	@echo "## library compiled successfully"
 
 $(K2PFTS_SRC) :
@@ -1789,15 +1431,15 @@
 
 # library libtess_fits
 #-----------------------------
-$(TFTS_LIB) : $(TFTS_OBJ)
+$(TFTS_LIB) : $(TFTS_OBJ) make_directories
 	@echo "***** creating library $(TFTS_CLAU) *****"
 	ar -rc $@ $(TFTS_OBJ)
 	ranlib $@
 	@echo "## library created successfully"
 
-$(TFTS_OBJ) : $(TFTS_SRC) $(TFTS_HDR_LIST) 
+$(TFTS_OBJ) : $(TFTS_SRC) $(TFTS_HDR_LIST)  make_directories
 	@echo "***** compiling $(TFTS_CLAU) *****"
-	$(COMPILADOR) -o $@ -c $(TFTS_SRC) $(TFTS_INC)
+	$(COMPILADOR) -o $@ -c $(TFTS_SRC) $(TFTS_INC) -I$(JUANFITS_PATH)
 	@echo "## library compiled successfully"
 
 $(TFTS_SRC) :
